// @shopify/hydrogen/dist/production/index.cjs
'use strict';

var hydrogenReact = require('@shopify/hydrogen-react');
var react = require('react');
var react$1 = require('@remix-run/react');
var jsxRuntime = require('react/jsx-runtime');
var cookie = require('worktop/cookie');
var na = require('content-security-policy-builder');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var na__default = /*#__PURE__*/_interopDefault(na);

var So=Object.defineProperty;var To=(e,t)=>()=>(e&&(t=e(e=0)),t);var vo=(e,t)=>{for(var r in t)So(e,r,{get:t[r],enumerable:!0});};var hr={};vo(hr,{default:()=>mr,logSeoTags:()=>gr});function mr({headTags:e}){return gr(e),null}function gr(e){console.log(" "),console.log("%cSEO Meta Tags",`${Ko}`),console.log(" "),e.forEach(t=>{if(t.tag==="script"){if(console.log("%c\u2022 JSON LD ",vt),t.children)try{console.table(JSON.parse(t.children),["name","content"]);}catch{console.log(t.children);}}else {if(console.log(`%c\u2022 ${t.tag} `,vt),t.children)if(typeof t.children=="string")console.log(`\u21B3 ${t.children}`);else try{Object.entries(JSON.parse(t.children)).map(([r,o])=>console.log(`\u21B3 ${o}`));}catch{console.log(t.children);}if(t.props.property==="og:image:url"){let r=t.props.content;zo(r).then(o=>{let n=`font-size: 400px; padding: 10px; background: white url(${o}) no-repeat center; background-size: contain;`;console.log("%c\u2022 Share image preview",vt),console.log("%c  ",n),console.log(`\u21B3 ${r}`);}).catch(o=>{console.error(o);});}Object.entries(t.props).map(([r,o])=>{console.log(`\u21B3 ${r} \u2192 ${o}`);});}console.log(" ");});}async function zo(e){let o=await(await(await fetch(e)).blob()).arrayBuffer();return `data:image/png;base64,${Jo(o)}`}function Jo(e){let t="",r=new Uint8Array(e),o=r.byteLength;for(let n=0;n<o;n++)t+=String.fromCharCode(r[n]);return btoa(t)}var vt,Ko,Cr=To(()=>{vt="text-transform: uppercase;",Ko="text-transform: uppercase; font-weight: bold; text-transform: uppercase;font-weight: bold";});var _e="public",Io="private",rt="no-store",Xt={maxAge:"max-age",staleWhileRevalidate:"stale-while-revalidate",sMaxAge:"s-maxage",staleIfError:"stale-if-error"};function Pe(e){let t=[];return Object.keys(e).forEach(r=>{r==="mode"?t.push(e[r]):Xt[r]&&t.push(`${Xt[r]}=${e[r]}`);}),t.join(", ")}function ot(){return {mode:rt}}function nt(e){if(e?.mode&&e?.mode!==_e&&e?.mode!==Io)throw Error("'mode' must be either 'public' or 'private'")}function oe(e){return nt(e),{mode:_e,maxAge:1,staleWhileRevalidate:9,...e}}function at(e){return nt(e),{mode:_e,maxAge:3600,staleWhileRevalidate:82800,...e}}function ce(e){return nt(e),{mode:_e,maxAge:1,staleWhileRevalidate:86399,...e}}function st(e){return e}function H(e){return String(e).includes("__proto__")?JSON.parse(e,Eo):JSON.parse(e)}function Eo(e,t){if(e!=="__proto__")return t}function Ue(e,t){return e&&t?{...e,...t}:e||ce()}function it(e){return Pe(Ue(e))}async function Ro(e,t){if(!e)return;let r=await e.match(t);if(!r){return}return r}async function wo(e,t,r,o){if(!e)return;let n=Ue(o),a=it(Ue(n,{maxAge:(n.maxAge||0)+(n.staleWhileRevalidate||0)})),s=it(Ue(n));r.headers.set("cache-control",a),r.headers.set("real-cache-control",s),r.headers.set("cache-put-date",String(Date.now())),await e.put(t,r);}async function bo(e,t){e&&await e.delete(t);}function Oo(e,t){let r=e.headers.get("real-cache-control"),o=0;if(r){let a=r.match(/max-age=(\d*)/);a&&a.length>1&&(o=parseFloat(a[1]));}return [(Date.now()-Number(t))/1e3,o]}function xo(e,t){let r=t.headers.get("cache-put-date");if(!r)return !1;let[o,n]=Oo(t,r),a=o>n;return a}var Le={get:Ro,set:wo,delete:bo,generateDefaultCacheControlHeader:it,isStale:xo};function Ae(e){return `https://shopify.dev/?${e}`}function Do(e){return e||ce()}async function Zt(e,t){if(!e)return;let r=Ae(t),o=new Request(r),n=await Le.get(e,o);if(!n)return;let a=await n.text();try{return [H(a),n]}catch{return [a,n]}}async function er(e,t,r,o){if(!e)return;let n=Ae(t),a=new Request(n),s=new Response(JSON.stringify(r));await Le.set(e,a,s,Do(o));}function tr(e,t){return Le.isStale(new Request(Ae(e)),t)}function rr(e){let t=Array.isArray(e)?e:[e],r="";for(let o of t)o!=null&&(typeof o=="object"?r+=JSON.stringify(o):r+=o.toString());return encodeURIComponent(r)}var ct=new Set;async function ke(e,t,{strategy:r=oe(),cacheInstance:o,shouldCacheResult:n=()=>!0,waitUntil:a,debugInfo:s}){let i=rr([...typeof e=="string"?[e]:e]),y=f=>{({displayName:f.displayName,url:f.response?.url,responseInit:{status:f.response?.status||0,statusText:f.response?.statusText||"",headers:Array.from(f.response?.headers.entries()||[])}});},l=void 0;if(!o||!r||r.mode===rt){let f=await t({addDebugData:y});return f}let d=f=>er(o,i,{value:f,debugInfo:void 0},r),g=await Zt(o,i);if(g&&typeof g[0]!="string"){let[{value:f,debugInfo:h},A]=g;let b=tr(i,A)?"STALE":"HIT";if(!ct.has(i)&&b==="STALE"){ct.add(i);let _=Promise.resolve().then(async()=>{let k=Date.now();try{let x=await t({addDebugData:y});n(x)&&(await d(x),l?.({result:x,cacheStatus:"PUT",overrideStartTime:k}));}catch(x){x.message&&(x.message="SWR in sub-request failed: "+x.message),console.error(x);}finally{ct.delete(i);}});a?.(_);}return f}let P=await t({addDebugData:y});if(n(P)){let f=Promise.resolve().then(async()=>{await d(P);});a?.(f);}return P}function _o(e,t){return [e,{status:t.status,statusText:t.statusText,headers:Array.from(t.headers.entries())}]}function or([e,t]){return [e,new Response(e,t)]}async function Ne(e,t,{cacheInstance:r,cache:o,cacheKey:n=[e,t],shouldCacheResponse:a,waitUntil:s,debugInfo:c}){return !o&&(!t.method||t.method==="GET")&&(o=oe()),ke(n,async()=>{let i=await fetch(e,t);if(!i.ok)return i;let u=await i.text().catch(()=>"");try{u&&(u=H(u));}catch{}return _o(u,i)},{cacheInstance:r,waitUntil:s,strategy:o??null,debugInfo:c,shouldCacheResult:i=>"ok"in i?!1:a(...or(i))}).then(i=>"ok"in i?[null,i]:or(i))}var ut="Custom-Storefront-Request-Group-ID",pt="X-Shopify-Storefront-Access-Token",dt="X-SDK-Variant",lt="X-SDK-Variant-Source",yt="X-SDK-Version";function nr(){return typeof crypto<"u"&&crypto.randomUUID?crypto.randomUUID():`weak-${Math.random().toString(16).substring(2)}`}var ar=new Set,Me=e=>{ar.has(e)||(console.warn(e),ar.add(e));},sr=new Set,ft=e=>{sr.has(e)||(console.error(new Error(e)),sr.add(e));};var Se="2025.1.0";function ue(e){return e.replace(/\s*#.*$/gm,"").replace(/\s+/gm," ").trim()}var Uo=/(^|}\s)query[\s({]/im,Lo=/(^|}\s)mutation[\s({]/im;function Ve(e,t){if(!Uo.test(e))throw new Error(`[h2:error:${t}] Can only execute queries`)}function Fe(e,t){if(!Lo.test(e))throw new Error(`[h2:error:${t}] Can only execute mutations`)}var ne=class extends Error{locations;path;extensions;constructor(t,r={}){let n=(r.clientOperation?`[h2:error:${r.clientOperation}] `:"")+t+(r.requestId?` - Request ID: ${r.requestId}`:"");super(n),this.name="GraphQLError",this.extensions=r.extensions,this.locations=r.locations,this.path=r.path,this.stack=r.stack||void 0;try{this.cause=JSON.stringify({...typeof r.cause=="object"?r.cause:{},requestId:r.requestId});}catch{r.cause&&(this.cause=r.cause);}}get[Symbol.toStringTag](){return this.name}toString(){let t=`${this.name}: ${this.message}`;if(this.path)try{t+=` | path: ${JSON.stringify(this.path)}`;}catch{}if(this.extensions)try{t+=` | extensions: ${JSON.stringify(this.extensions)}`;}catch{}return t+=`
`,this.stack&&(t+=`${this.stack.slice(this.stack.indexOf(`
`)+1)}
`),t}toJSON(){return {name:"Error",message:""}}};function Te({url:e,response:t,errors:r,type:o,query:n,queryVariables:a,ErrorConstructor:s=Error,client:c="storefront"}){let i=(typeof r=="string"?r:r?.map?.(p=>p.message).join(`
`))||`URL: ${e}
API response error: ${t.status}`,u=new ne(i,{query:n,queryVariables:a,cause:{errors:r},clientOperation:`${c}.${o}`,requestId:t.headers.get("x-request-id")});throw new s(u.message,{cause:u.cause})}function pe(e,t={}){let r=new Error,o=(n,a="Error")=>{let s=(r.stack??"").split(`
`).slice(3+(t.stackOffset??0)).join(`
`).replace(/ at loader(\d+) \(/,(c,i)=>c.replace(i,""));return `${a}: ${n}
`+s};return e.then(n=>{if(n?.errors&&Array.isArray(n.errors)){let a=typeof t.logErrors=="function"?t.logErrors:()=>t.logErrors??!1;n.errors.forEach(s=>{s&&(s.stack=o(s.message,s.name),a(s)&&console.error(s));});}return n}).catch(n=>{throw n&&(n.stack=o(n.message,n.name)),n})}var $=void 0;var qo={language:"EN",country:"US"};function pr(e){let {storefrontHeaders:t,cache:r,waitUntil:o,i18n:n,storefrontId:a,logErrors:s=!0,...c}=e,{getPublicTokenHeaders:u,getPrivateTokenHeaders:p,getStorefrontApiUrl:y,getShopifyDomain:m}=hydrogenReact.createStorefrontClient(c),d=(c.privateStorefrontToken?p:u)({contentType:"json",buyerIp:t?.buyerIp||""});if(d[ut]=t?.requestGroupId||nr(),a&&(d[hydrogenReact.SHOPIFY_STOREFRONT_ID_HEADER]=a),(d["user-agent"]=`Hydrogen ${Se}`),t&&t.cookie){let f=hydrogenReact.getShopifyCookies(t.cookie??"");f[hydrogenReact.SHOPIFY_Y]&&(d[hydrogenReact.SHOPIFY_STOREFRONT_Y_HEADER]=f[hydrogenReact.SHOPIFY_Y]),f[hydrogenReact.SHOPIFY_S]&&(d[hydrogenReact.SHOPIFY_STOREFRONT_S_HEADER]=f[hydrogenReact.SHOPIFY_S]);}let g=JSON.stringify({"content-type":d["content-type"],"user-agent":d["user-agent"],[dt]:d[dt],[lt]:d[lt],[yt]:d[yt],[pt]:d[pt]});async function P({query:f,mutation:h,variables:A,cache:b,headers:_=[],storefrontApiVersion:k,displayName:x,stackInfo:K}){let he=_ instanceof Headers?Object.fromEntries(_.entries()):Array.isArray(_)?Object.fromEntries(_):_,te=f??h,z={...A};n&&(!A?.country&&/\$country/.test(te)&&(z.country=n.country),!A?.language&&/\$language/.test(te)&&(z.language=n.language));let ie=y({storefrontApiVersion:k}),Ce=JSON.stringify({query:te,variables:z}),C={method:"POST",headers:{...d,...he},body:Ce},S=[ie,C.method,g,C.body],[v,I]=await Ne(ie,C,{cacheInstance:h?void 0:r,cache:b||ce(),cacheKey:S,waitUntil:o,shouldCacheResponse:D=>!D?.errors,debugInfo:{requestId:C.headers[ut],displayName:x,url:ie,stackInfo:K,graphql:Ce,purpose:t?.purpose}}),q={url:ie,response:I,type:h?"mutation":"query",query:te,queryVariables:z,errors:void 0};if(!I.ok){let D,O=v;try{O??=await I.text(),D=H(O);}catch{D=[{message:O??"Could not parse Storefront API response"}];}Te({...q,errors:D});}let{data:re,errors:N}=v,F=N?.map(({message:D,...O})=>new ne(D,{...O,clientOperation:`storefront.${q.type}`,requestId:I.headers.get("x-request-id"),queryVariables:z,query:te}));return T(re,F)}return {storefront:{query(f,h){f=ue(f),Ve(f,"storefront.query");let A=ur?.(f);return pe(P({...h,query:f,stackInfo:$?.(A)}),{stackOffset:A,logErrors:s})},mutate(f,h){f=ue(f),Fe(f,"storefront.mutate");let A=ur?.(f);return pe(P({...h,mutation:f,stackInfo:$?.(A)}),{stackOffset:A,logErrors:s})},cache:r,CacheNone:ot,CacheLong:at,CacheShort:oe,CacheCustom:st,generateCacheControlHeader:Pe,getPublicTokenHeaders:u,getPrivateTokenHeaders:p,getShopifyDomain:m,getApiUrl:y,i18n:n??qo}}}var ur=void 0;function T(e,t){return {...e,...t&&{errors:t}}}function B(e,t){return mt(e.headers,t)}function mt(e,t){let r=e?.get?.(t)??e?.[t];return typeof r=="string"?r:null}function Y(e){return {requestId:e?B(e,"request-id"):void 0,purpose:e?B(e,"purpose"):void 0}}function $o(e){let{cache:t,waitUntil:r,request:o}=e;return {run:({cacheKey:n,cacheStrategy:a,shouldCacheResult:s},c)=>ke(n,c,{shouldCacheResult:s,strategy:a,cacheInstance:t,waitUntil:r,debugInfo:{...Y(o),stackInfo:$?.()}}),fetch:(n,a,s)=>Ne(n,a??{},{waitUntil:r,cacheKey:[n,a],cacheInstance:t,debugInfo:{url:n,...Y(o),stackInfo:$?.(),displayName:s?.displayName},cache:s.cacheStrategy,...s}).then(([c,i])=>({data:c,response:i}))}}var gt=class{#e;constructor(){this.#e=new Map;}add(t){throw new Error("Method not implemented. Use `put` instead.")}addAll(t){throw new Error("Method not implemented. Use `put` instead.")}matchAll(t,r){throw new Error("Method not implemented. Use `match` instead.")}async put(t,r){if(t.method!=="GET")throw new TypeError("Cannot cache response to non-GET request.");if(r.status===206)throw new TypeError("Cannot cache response to a range request (206 Partial Content).");if(r.headers.get("vary")?.includes("*"))throw new TypeError("Cannot cache response with 'Vary: *' header.");this.#e.set(t.url,{body:new Uint8Array(await r.arrayBuffer()),status:r.status,headers:[...r.headers],timestamp:Date.now()});}async match(t){if(t.method!=="GET")return;let r=this.#e.get(t.url);if(!r)return;let{body:o,timestamp:n,...a}=r,s=new Headers(a.headers),c=s.get("cache-control")||s.get("real-cache-control")||"",i=parseInt(c.match(/max-age=(\d+)/)?.[1]||"0",10),u=parseInt(c.match(/stale-while-revalidate=(\d+)/)?.[1]||"0",10),p=(Date.now()-n)/1e3;if(p>i+u){this.#e.delete(t.url);return}let m=p>i;return s.set("cache",m?"STALE":"HIT"),s.set("date",new Date(n).toUTCString()),new Response(o,{status:a.status??200,headers:s})}async delete(t){return this.#e.has(t.url)?(this.#e.delete(t.url),!0):!1}keys(t){let r=[];for(let o of this.#e.keys())(!t||t.url===o)&&r.push(new Request(o));return Promise.resolve(r)}};function qe(e){if(!e)return;let{pathname:t,search:r}=new URL(e),o=t+r,n=new URLSearchParams(r),a=n.get("return_to")||n.get("redirect");if(a){if(lr(e,a))return a;console.warn(`Cross-domain redirects are not supported. Tried to redirect from ${o} to ${a}`);}}function lr(e,t){try{return new URL(e).origin===new URL(t,e).origin}catch{return !1}}function ht({requestUrl:e,defaultUrl:t,redirectUrl:r}){let o=e,n=dr(e,t),a=r?dr(e,r):n;return lr(e,a.toString())?a.toString():(console.warn(`Cross-domain redirects are not supported. Tried to redirect from ${o} to ${a}. Default url ${n} is used instead.`),n.toString())}function dr(e,t){return Ho(t)?new URL(t):new URL(t,new URL(e).origin)}function Ho(e){try{return new URL(e),!0}catch{return !1}}async function Bo(e){let{storefront:t,request:r,noAdminRedirect:o,matchQueryParams:n,response:a=new Response("Not Found",{status:404})}=e,s=new URL(r.url),{pathname:c,searchParams:i}=s,u=i.has("_data");i.delete("redirect"),i.delete("return_to"),i.delete("_data");let p=(n?s.toString().replace(s.origin,""):c).toLowerCase();if(s.pathname==="/admin"&&!o)return Pt(`${t.getShopifyDomain()}/admin`,u,i,n);try{let{urlRedirects:y}=await t.query(Go,{variables:{query:"path:"+p.replace(/\/+$/,"")}}),m=y?.edges?.[0]?.node?.target;if(m)return Pt(m,u,i,n);let l=qe(r.url);if(l)return Pt(l,u,i,n)}catch(y){console.error(`Failed to fetch redirects from Storefront API for route ${p}`,y);}return a}var Ct="https://example.com";function Pt(e,t,r,o){let n=new URL(e,Ct);if(!o)for(let[a,s]of r)n.searchParams.append(a,s);return t?new Response(null,{status:200,headers:{"X-Remix-Redirect":n.toString().replace(Ct,""),"X-Remix-Status":"301"}}):new Response(null,{status:301,headers:{location:n.toString().replace(Ct,"")}})}var Go=`#graphql
  query redirects($query: String) {
    urlRedirects(first: 1, query: $query) {
      edges {
        node {
          target
        }
      }
    }
  }
`;var Qo=async function({request:t,context:r}){let o=r.storefront,n=r.customerAccount,a=new URL(t.url);if(!o)throw new Error("GraphiQL: Hydrogen's storefront client must be injected in the loader context.");let s={};if(o){let u="X-Shopify-Storefront-Access-Token";s.storefront={name:"Storefront API",authHeader:u,accessToken:o.getPublicTokenHeaders()[u],apiUrl:o.getApiUrl(),icon:"SF"};}if(n){let u=await(await fetch(a.origin+"/graphiql/customer-account.schema.json")).json(),p=await n.getAccessToken();u&&(s["customer-account"]={name:"Customer Account API",value:u,authHeader:"Authorization",accessToken:p,apiUrl:n.getApiUrl(),icon:"CA"});}let c="https://avatars.githubusercontent.com/u/12972006?s=48&v=4",i=String.raw;return new Response(i`
      <!DOCTYPE html>
      <html lang="en">
        <head>
          <title>GraphiQL</title>
          <link rel="icon" type="image/x-icon" href="${c}" />
          <style>
            body {
              height: 100%;
              margin: 0;
              width: 100%;
              overflow: hidden;
              background-color: hsl(219, 29%, 18%);
            }

            #graphiql {
              height: 100vh;
            }

            #graphiql > .placeholder {
              color: slategray;
              width: fit-content;
              margin: 40px auto;
              font-family: Arial;
            }

            .graphiql-api-toolbar-label {
              position: absolute;
              bottom: -6px;
              right: -4px;
              font-size: 8px;
            }
          </style>

          <script
            crossorigin
            src="https://unpkg.com/react@18/umd/react.development.js"
          ></script>
          <script
            crossorigin
            src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"
          ></script>
          <link
            rel="stylesheet"
            href="https://unpkg.com/graphiql@3/graphiql.min.css"
          />
          <link
            rel="stylesheet"
            href="https://unpkg.com/@graphiql/plugin-explorer/dist/style.css"
          />
        </head>

        <body>
          <div id="graphiql">
            <div class="placeholder">Loading GraphiQL...</div>
          </div>

          <script
            src="https://unpkg.com/graphiql@3/graphiql.min.js"
            type="application/javascript"
            crossorigin="anonymous"
          ></script>
          <script
            src="https://unpkg.com/@graphiql/plugin-explorer/dist/index.umd.js"
            type="application/javascript"
            crossorigin="anonymous"
          ></script>

          <script>
            const windowUrl = new URL(document.URL);
            const startingSchemaKey =
              windowUrl.searchParams.get('schema') || 'storefront';

            let query = '{ shop { name } }';
            if (windowUrl.searchParams.has('query')) {
              query = decodeURIComponent(
                windowUrl.searchParams.get('query') ?? query,
              );
            }

            // Prettify query
            query = GraphiQL.GraphQL.print(GraphiQL.GraphQL.parse(query));

            let variables;
            if (windowUrl.searchParams.has('variables')) {
              variables = decodeURIComponent(
                windowUrl.searchParams.get('variables') ?? '',
              );
            }

            // Prettify variables
            if (variables) {
              variables = JSON.stringify(JSON.parse(variables), null, 2);
            }

            const schemas = ${JSON.stringify(s)};
            let lastActiveTabIndex = -1;
            let lastTabAmount = -1;

            const root = ReactDOM.createRoot(
              document.getElementById('graphiql'),
            );

            root.render(React.createElement(RootWrapper));

            const TAB_STATE_KEY = 'graphiql:tabState';
            const storage = {
              getTabState: () =>
                JSON.parse(localStorage.getItem(TAB_STATE_KEY)),
              setTabState: (state) =>
                localStorage.setItem(TAB_STATE_KEY, JSON.stringify(state)),
            };

            let nextSchemaKey;

            function RootWrapper() {
              const [activeSchema, setActiveSchema] =
                React.useState(startingSchemaKey);

              const schema = schemas[activeSchema];
              if (!schema) {
                throw new Error('No schema found for ' + activeSchema);
              }

              const keys = Object.keys(schemas);

              return React.createElement(
                GraphiQL,
                {
                  fetcher: GraphiQL.createFetcher({
                    url: schema.apiUrl,
                    headers: {[schema.authHeader]: schema.accessToken},
                  }),
                  defaultEditorToolsVisibility: true,
                  query,
                  variables,
                  schema: schema.value,
                  plugins: [GraphiQLPluginExplorer.explorerPlugin()],
                  onTabChange: (state) => {
                    const {activeTabIndex, tabs} = state;
                    const activeTab = tabs[activeTabIndex];

                    if (
                      activeTabIndex === lastActiveTabIndex &&
                      lastTabAmount === tabs.length
                    ) {
                      if (
                        nextSchemaKey &&
                        activeTab &&
                        activeTab.schemaKey !== nextSchemaKey
                      ) {
                        activeTab.schemaKey = nextSchemaKey;
                        nextSchemaKey = undefined;

                        // Sync state to localStorage. GraphiQL resets the state
                        // asynchronously, so we need to do it in a timeout.
                        storage.setTabState(state);
                        setTimeout(() => storage.setTabState(state), 500);
                      }

                      // React rerrendering, skip
                      return;
                    }

                    if (activeTab) {
                      if (!activeTab.schemaKey) {
                        // Creating a new tab
                        if (lastTabAmount < tabs.length) {
                          activeTab.schemaKey = activeSchema;
                          storage.setTabState(state);
                        }
                      }

                      const nextSchema = activeTab.schemaKey || 'storefront';

                      if (nextSchema !== activeSchema) {
                        setActiveSchema(nextSchema);
                      }
                    }

                    lastActiveTabIndex = activeTabIndex;
                    lastTabAmount = tabs.length;
                  },
                  toolbar: {
                    additionalComponent: function () {
                      const schema = schemas[activeSchema];

                      return React.createElement(
                        GraphiQL.React.ToolbarButton,
                        {
                          onClick: () => {
                            const activeKeyIndex = keys.indexOf(activeSchema);
                            nextSchemaKey =
                              keys[(activeKeyIndex + 1) % keys.length];

                            // This triggers onTabChange
                            if (nextSchemaKey) setActiveSchema(nextSchemaKey);
                          },
                          label: 'Toggle between different API schemas',
                        },
                        React.createElement(
                          'div',
                          {
                            key: 'api-wrapper',
                            className: 'graphiql-toolbar-icon',
                            style: {position: 'relative', fontWeight: 'bolder'},
                          },
                          [
                            React.createElement(
                              'div',
                              {key: 'icon', style: {textAlign: 'center'}},
                              [
                                schema.icon,
                                React.createElement(
                                  'div',
                                  {
                                    key: 'icon-label',
                                    className: 'graphiql-api-toolbar-label',
                                  },
                                  'API',
                                ),
                              ],
                            ),
                          ],
                        ),
                      );
                    },
                  },
                },
                [
                  React.createElement(
                    GraphiQL.Logo,
                    {
                      key: 'Logo replacement',
                    },
                    [
                      React.createElement(
                        'div',
                        {
                          key: 'Logo wrapper',
                          style: {display: 'flex', alignItems: 'center'},
                        },
                        [
                          React.createElement(
                            'div',
                            {
                              key: 'api',
                              className: 'graphiql-logo',
                              style: {
                                paddingRight: 0,
                                whiteSpace: 'nowrap',
                              },
                            },
                            [schema.name],
                          ),
                          React.createElement(GraphiQL.Logo, {key: 'logo'}),
                        ],
                      ),
                    ],
                  ),
                ],
              );
            }
          </script>
        </body>
      </html>
    `,{status:200,headers:{"content-type":"text/html"}})};var Wo={"&":"\\u0026",">":"\\u003e","<":"\\u003c","\u2028":"\\u2028","\u2029":"\\u2029"},jo=/[&><\u2028\u2029]/g;function yr(e){return e.replace(jo,t=>Wo[t])}var X="Error in SEO input: ",G={title:{validate:e=>{if(typeof e!="string")throw new Error(X.concat("`title` should be a string"));if(typeof e=="string"&&e.length>120)throw new Error(X.concat("`title` should not be longer than 120 characters"));return e}},description:{validate:e=>{if(typeof e!="string")throw new Error(X.concat("`description` should be a string"));if(typeof e=="string"&&e.length>155)throw new Error(X.concat("`description` should not be longer than 155 characters"));return e}},url:{validate:e=>{if(typeof e!="string")throw new Error(X.concat("`url` should be a string"));if(typeof e=="string"&&!e.startsWith("http"))throw new Error(X.concat("`url` should be a valid URL"));return e}},handle:{validate:e=>{if(typeof e!="string")throw new Error(X.concat("`handle` should be a string"));if(typeof e=="string"&&!e.startsWith("@"))throw new Error(X.concat("`handle` should start with `@`"));return e}}};function fr(e){let t=[];for(let r of Object.keys(e))switch(r){case"title":{let o=Q(G.title,e.title),n=St(e?.titleTemplate,o);if(!n)break;t.push(U("title",{title:n}),U("meta",{property:"og:title",content:n}),U("meta",{name:"twitter:title",content:n}));break}case"description":{let o=Q(G.description,e.description);if(!o)break;t.push(U("meta",{name:"description",content:o}),U("meta",{property:"og:description",content:o}),U("meta",{name:"twitter:description",content:o}));break}case"url":{let o=Q(G.url,e.url);if(!o)break;let a=o.split("?")[0].replace(/\/$/,"");t.push(U("link",{rel:"canonical",href:a}),U("meta",{property:"og:url",content:a}));break}case"handle":{let o=Q(G.handle,e.handle);if(!o)break;t.push(U("meta",{name:"twitter:site",content:o}),U("meta",{name:"twitter:creator",content:o}));break}case"media":{let o,n=Z(e.media);for(let a of n)if(typeof a=="string"&&t.push(U("meta",{name:"og:image",content:a})),a&&typeof a=="object"){let s=a.type||"image",c=a?{url:a?.url,secure_url:a?.url,type:Tt(a.url),width:a?.width,height:a?.height,alt:a?.altText}:{};for(let i of Object.keys(c))c[i]&&(o=c[i],t.push(U("meta",{property:`og:${s}:${i}`,content:o},c.url)));}break}case"jsonLd":{let o=Z(e.jsonLd),n=0;for(let a of o){if(typeof a!="object")continue;let s=U("script",{type:"application/ld+json",children:JSON.stringify(a,(c,i)=>typeof i=="string"?yr(i):i)},`json-ld-${a?.["@type"]||a?.name||n++}`);t.push(s);}break}case"alternates":{let o=Z(e.alternates);for(let n of o){if(!n)continue;let{language:a,url:s,default:c}=n,i=a?`${a}${c?"-default":""}`:void 0;t.push(U("link",{rel:"alternate",hrefLang:i,href:s}));}break}case"robots":{if(!e.robots)break;let{maxImagePreview:o,maxSnippet:n,maxVideoPreview:a,noArchive:s,noFollow:c,noImageIndex:i,noIndex:u,noSnippet:p,noTranslate:y,unavailableAfter:m}=e.robots,l=[s&&"noarchive",i&&"noimageindex",p&&"nosnippet",y&&"notranslate",o&&`max-image-preview:${o}`,n&&`max-snippet:${n}`,a&&`max-video-preview:${a}`,m&&`unavailable_after:${m}`],d=(u?"noindex":"index")+","+(c?"nofollow":"follow");for(let g of l)g&&(d+=`,${g}`);t.push(U("meta",{name:"robots",content:d}));break}}return t.flat().sort((r,o)=>r.key.localeCompare(o.key))}function U(e,t,r){let o={tag:e,props:{},key:""};return e==="title"?(o.children=t.title,o.key=At(o),o):e==="script"?(o.children=typeof t.children=="string"?t.children:"",o.key=At(o,r),delete t.children,o.props=t,o):(o.props=t,Object.keys(o.props).forEach(n=>!o.props[n]&&delete o.props[n]),o.key=At(o,r),o)}function At(e,t){let{tag:r,props:o}=e;if(r==="title")return "0-title";if(r==="meta"){let n=o.content===t&&typeof o.property=="string"&&!o.property.endsWith("secure_url")&&"0";return [r,...[t,n],o.property||o.name].filter(s=>s).join("-")}return r==="link"?[r,o.rel,o.hrefLang||o.media].filter(a=>a).join("-").replace(/\s+/g,"-"):r==="script"?`${r}-${t}`:`${r}-${o.type}`}function St(e,t){if(t)return e?typeof e=="function"?e(t):e.replace("%s",t??""):t}function Tt(e){switch(e&&e.split(".").pop()){case"svg":return "image/svg+xml";case"png":return "image/png";case"gif":return "image/gif";case"swf":return "application/x-shockwave-flash";case"mp3":return "audio/mpeg";case"jpg":case"jpeg":default:return "image/jpeg"}}function Z(e){return Array.isArray(e)?e:[e]}function Q(e,t){try{return e.validate(t)}catch(r){return console.warn(r.message),t}}var rn=react.lazy(()=>Promise.resolve().then(()=>(Cr(),hr)));function on({debug:e}){let t=react$1.useMatches(),r=react$1.useLocation();console.warn("[h2:warn:Seo] The `<Seo/>` component is deprecated. Use `getSeoMeta` instead.\nSee: https://shopify.dev/docs/api/hydrogen/2025-01/utilities/getseometa");let o=react.useMemo(()=>t.flatMap(s=>{let{handle:c,...i}=s,u={...i,...r},p=c?.seo,y=i?.data?.seo;return !p&&!y?[]:p?$e(p,u):[y]}).reduce((s,c)=>{Object.keys(c).forEach(u=>!c[u]&&delete c[u]);let{jsonLd:i}=c;return i?s?.jsonLd?Array.isArray(i)?{...s,...c,jsonLd:[...s.jsonLd,...i]}:{...s,...c,jsonLd:[...s.jsonLd,i]}:{...s,...c,jsonLd:[i]}:{...s,...c}},{}),[t,r]),{html:n,loggerMarkup:a}=react.useMemo(()=>{let s=fr(o),c=s.map(u=>u.tag==="script"?react.createElement(u.tag,{...u.props,key:u.key,dangerouslySetInnerHTML:{__html:u.children}}):react.createElement(u.tag,{...u.props,key:u.key},u.children)),i=react.createElement(react.Suspense,{fallback:null},react.createElement(rn,{headTags:s}));return {html:c,loggerMarkup:i}},[o]);return react.createElement(react.Fragment,null,n,e&&a)}function $e(e,...t){if(e instanceof Function)return $e(e(...t),...t);let r={};return Array.isArray(e)?(r=e.reduce((o,n)=>[...o,$e(n)],[]),r):e instanceof Object?(Object.entries(e).forEach(([n,a])=>{r[n]=$e(a,...t);}),r):e}function nn(...e){let t=[],r=e.reduce((o,n)=>{if(!n)return o;Object.keys(n).forEach(s=>!n[s]&&delete n[s]);let{jsonLd:a}=n;return a?o?.jsonLd?{...o,...n,jsonLd:Z(o.jsonLd).concat(a)}:{...o,...n,jsonLd:[a]}:{...o,...n}},{})||{};for(let o of Object.keys(r))switch(o){case"title":{let n=Q(G.title,r.title),a=St(r?.titleTemplate,n);if(!a)break;t.push({title:a},{property:"og:title",content:a},{property:"twitter:title",content:a});break}case"description":{let n=Q(G.description,r.description);if(!n)break;t.push({name:"description",content:n},{property:"og:description",content:n},{property:"twitter:description",content:n});break}case"url":{let n=Q(G.url,r.url);if(!n)break;let s=n.split("?")[0].replace(/\/$/,"");t.push({tagName:"link",rel:"canonical",href:s},{property:"og:url",content:s});break}case"handle":{let n=Q(G.handle,r.handle);if(!n)break;t.push({property:"twitter:site",content:n},{property:"twitter:creator",content:n});break}case"media":{let n,a=Z(r.media);for(let s of a)if(typeof s=="string"&&t.push({property:"og:image",content:s}),s&&typeof s=="object"){let c=s.type||"image",i=s?{url:s?.url,secure_url:s?.url,type:Tt(s.url),width:s?.width,height:s?.height,alt:s?.altText}:{};for(let u of Object.keys(i))i[u]&&(n=i[u],t.push({property:`og:${c}:${u}`,content:n}));}break}case"jsonLd":{let n=Z(r.jsonLd);for(let s of n)typeof s!="object"||Object.keys(s).length===0||t.push({"script:ld+json":s});break}case"alternates":{let n=Z(r.alternates);for(let a of n){if(!a)continue;let{language:s,url:c,default:i}=a,u=s?`${s}${i?"-default":""}`:void 0;t.push({tagName:"link",rel:"alternate",hrefLang:u,href:c});}break}case"robots":{if(!r.robots)break;let{maxImagePreview:n,maxSnippet:a,maxVideoPreview:s,noArchive:c,noFollow:i,noImageIndex:u,noIndex:p,noSnippet:y,noTranslate:m,unavailableAfter:l}=r.robots,d=[c&&"noarchive",u&&"noimageindex",y&&"nosnippet",m&&"notranslate",n&&`max-image-preview:${n}`,a&&`max-snippet:${a}`,s&&`max-video-preview:${s}`,l&&`unavailable_after:${l}`],g=(p?"noindex":"index")+","+(i?"nofollow":"follow");for(let P of d)P&&(g+=`,${P}`);t.push({name:"robots",content:g});break}}return t}function cn({connection:e,children:t=()=>(console.warn("<Pagination> requires children to work properly"),null),namespace:r=""}){let [o,n]=react.useState(!1),a=react$1.useNavigation(),s=react$1.useLocation();react$1.useNavigate();react.useEffect(()=>{a.state==="idle"&&n(!1);},[a.state]);let{endCursor:i,hasNextPage:u,hasPreviousPage:p,nextPageUrl:y,nodes:m,previousPageUrl:l,startCursor:d}=un(e,r),g=react.useMemo(()=>({...s.state,pagination:{...s.state?.pagination||{},[r]:{pageInfo:{endCursor:i,hasPreviousPage:p,hasNextPage:u,startCursor:d},nodes:m}}}),[i,u,p,d,m,r,s.state]),P=react.useMemo(()=>react.forwardRef(function(A,b){return u?react.createElement(react$1.Link,{preventScrollReset:!0,...A,to:y,state:g,replace:!0,ref:b,onClick:()=>n(!0)}):null}),[u,y,g]),f=react.useMemo(()=>react.forwardRef(function(A,b){return p?react.createElement(react$1.Link,{preventScrollReset:!0,...A,to:l,state:g,replace:!0,ref:b,onClick:()=>n(!0)}):null}),[p,l,g]);return t({state:g,hasNextPage:u,hasPreviousPage:p,isLoading:o,nextPageUrl:y,nodes:m,previousPageUrl:l,NextLink:P,PreviousLink:f})}function He(e,t){let r=new URLSearchParams(e);return Object.keys(t?.pagination||{}).forEach(n=>{let a=n===""?"":`${n}_`,s=`${a}cursor`,c=`${a}direction`;r.delete(s),r.delete(c);}),r.toString()}function Ie(e){throw new Error(`The Pagination component requires ${"`"+e+"`"} to be a part of your query. See the guide on how to setup your query to include ${"`"+e+"`"}: https://shopify.dev/docs/custom-storefronts/hydrogen/data-fetching/pagination#setup-the-paginated-query`)}function un(e,t=""){e.pageInfo||Ie("pageInfo"),typeof e.pageInfo.startCursor>"u"&&Ie("pageInfo.startCursor"),typeof e.pageInfo.endCursor>"u"&&Ie("pageInfo.endCursor"),typeof e.pageInfo.hasNextPage>"u"&&Ie("pageInfo.hasNextPage"),typeof e.pageInfo.hasPreviousPage>"u"&&Ie("pageInfo.hasPreviousPage");let r=react$1.useNavigation(),o=react$1.useNavigate(),{state:n,search:a,pathname:s}=react$1.useLocation(),c=t?`${t}_cursor`:"cursor",i=t?`${t}_direction`:"direction",y=new URLSearchParams(a).get(i)==="previous",m=react.useMemo(()=>!globalThis?.window?.__hydrogenHydrated||!n?.pagination?.[t]?.nodes?hydrogenReact.flattenConnection(e):y?[...hydrogenReact.flattenConnection(e),...n.pagination[t].nodes||[]]:[...n.pagination[t].nodes||[],...hydrogenReact.flattenConnection(e)],[n,e,t]),l=react.useMemo(()=>{let f=globalThis?.window?.__hydrogenHydrated,h=n?.pagination?.[t]?.pageInfo,A=!f||h?.startCursor===void 0?e.pageInfo.startCursor:h.startCursor,b=!f||h?.endCursor===void 0?e.pageInfo.endCursor:h.endCursor,_=!f||h?.hasPreviousPage===void 0?e.pageInfo.hasPreviousPage:h.hasPreviousPage,k=!f||h?.hasNextPage===void 0?e.pageInfo.hasNextPage:h.hasNextPage;return n?.pagination?.[t]?.nodes&&(y?(A=e.pageInfo.startCursor,_=e.pageInfo.hasPreviousPage):(b=e.pageInfo.endCursor,k=e.pageInfo.hasNextPage)),{startCursor:A,endCursor:b,hasPreviousPage:_,hasNextPage:k}},[y,n,t,e.pageInfo.hasNextPage,e.pageInfo.hasPreviousPage,e.pageInfo.startCursor,e.pageInfo.endCursor]),d=react.useRef({params:He(a,n),pathname:s});react.useEffect(()=>{window.__hydrogenHydrated=!0;},[]),react.useEffect(()=>{let f=He(a,n),h=d.current.params;(s!==d.current.pathname||f!==h)&&!(r.state==="idle"&&!r.location)&&(d.current={pathname:s,params:He(a,n)},o(`${s}?${He(a,n)}`,{replace:!0,preventScrollReset:!0,state:{nodes:void 0,pageInfo:void 0}}));},[s,a,n]);let g=react.useMemo(()=>{let f=new URLSearchParams(a);return f.set(i,"previous"),l.startCursor&&f.set(c,l.startCursor),`?${f.toString()}`},[a,l.startCursor]),P=react.useMemo(()=>{let f=new URLSearchParams(a);return f.set(i,"next"),l.endCursor&&f.set(c,l.endCursor),`?${f.toString()}`},[a,l.endCursor]);return {...l,previousPageUrl:g,nextPageUrl:P,nodes:m}}function pn(e,t={pageBy:20}){if(typeof e?.url>"u")throw new Error("getPaginationVariables must be called with the Request object passed to your loader function");let{pageBy:r,namespace:o=""}=t,n=new URLSearchParams(new URL(e.url).search),a=o?`${o}_cursor`:"cursor",s=o?`${o}_direction`:"direction",c=n.get(a)??void 0;return (n.get(s)==="previous"?"previous":"next")==="previous"?{last:r,startCursor:c??null}:{first:r,endCursor:c??null}}var Be="2025-01",de=`Shopify Hydrogen ${Se}`,Rr="30243aa5-17c1-465a-8493-944bcc4e88aa",L="customerAccount",le="buyer";var V=class extends Response{constructor(t,r,o){super(`Bad request: ${t}`,{status:400,headers:o});}};function Re(e,t={}){let r=t.headers?new Headers(t.headers):new Headers({});return r.set("location",e),new Response(null,{status:t.status||302,headers:r})}async function dn({session:e,customerAccountId:t,customerAccountTokenExchangeUrl:r,httpsOrigin:o,debugInfo:n,exchangeForStorefrontCustomerAccessToken:a}){let s=new URLSearchParams,c=e.get(L),i=c?.refreshToken,u=c?.idToken;if(!i)throw new V("Unauthorized","No refreshToken found in the session. Make sure your session is configured correctly and passed to `createCustomerAccountClient`.");s.append("grant_type","refresh_token"),s.append("refresh_token",i),s.append("client_id",t);let p={"content-type":"application/x-www-form-urlencoded","User-Agent":de,Origin:o};new Date().getTime();let m=r,l=await fetch(m,{method:"POST",headers:p,body:s});if(!l.ok){let h=await l.text();throw new Response(h,{status:l.status,headers:{"Content-Type":"text/html; charset=utf-8"}})}let{access_token:d,expires_in:g,refresh_token:P}=await l.json(),f=await Rt(d,t,r,o);e.set(L,{accessToken:f,expiresAt:new Date(new Date().getTime()+(g-120)*1e3).getTime()+"",refreshToken:P,idToken:u}),await a();}function ye(e){e.unset(L),e.unset(le);}async function wr({locks:e,expiresAt:t,session:r,customerAccountId:o,customerAccountTokenExchangeUrl:n,httpsOrigin:a,debugInfo:s,exchangeForStorefrontCustomerAccessToken:c}){if(parseInt(t,10)-1e3<new Date().getTime())try{e.refresh||(e.refresh=dn({session:r,customerAccountId:o,customerAccountTokenExchangeUrl:n,httpsOrigin:a,debugInfo:s,exchangeForStorefrontCustomerAccessToken:c})),await e.refresh,delete e.refresh;}catch(i){throw ye(r),i&&i.status!==401?i:new V("Unauthorized","Login before querying the Customer Account API.")}}function br(){let e=ln();return xr(e)}async function Or(e){let t=await crypto.subtle.digest({name:"SHA-256"},new TextEncoder().encode(e)),r=yn(t);return xr(r)}function ln(){let e=new Uint8Array(32);return crypto.getRandomValues(e),String.fromCharCode.apply(null,Array.from(e))}function xr(e){return btoa(e).replace(/\+/g,"-").replace(/\//g,"_").replace(/=/g,"")}function yn(e){let t=new Uint8Array(e),r=Array.from(t);return String.fromCharCode(...r)}function Dr(){let e=Date.now().toString(),t=Math.random().toString(36).substring(2);return e+t}async function Rt(e,t,r,o,n){let a=t;if(!e)throw new V("Unauthorized","oAuth access token was not provided during token exchange.");let s=new URLSearchParams;s.append("grant_type","urn:ietf:params:oauth:grant-type:token-exchange"),s.append("client_id",a),s.append("audience",Rr),s.append("subject_token",e),s.append("subject_token_type","urn:ietf:params:oauth:token-type:access_token"),s.append("scopes","https://api.customers.com/auth/customer.graphql");let c={"content-type":"application/x-www-form-urlencoded","User-Agent":de,Origin:o};new Date().getTime();let u=r,p=await fetch(u,{method:"POST",headers:c,body:s});let y=await p.json();if(y.error)throw new V(y.error_description);return y.access_token}function _r(e){return fn(e).payload.nonce}function fn(e){let[t,r,o]=e.split("."),n=JSON.parse(atob(t)),a=JSON.parse(atob(r));return {header:n,payload:a,signature:o}}function Ge(){return gn(mn())}function mn(){try{return crypto.getRandomValues(new Uint8Array(16))}catch{return new Uint8Array(16).map(()=>Math.random()*255|0)}}function gn(e){return Array.from(e,function(t){return ("0"+(t&255).toString(16)).slice(-2)}).join("")}function Ur(e,t){let r=`https://shopify.com/${t}`,o=`https://shopify.com/authentication/${t}`;return function(a){switch(a){case"CA_BASE_URL":return r;case"CA_BASE_AUTH_URL":return o;case"GRAPHQL":return `${r}/account/customer/api/${e}/graphql`;case"AUTH":return `${o}/oauth/authorize`;case"LOGIN_SCOPE":return t?"openid email customer-account-api:full":"openid email https://api.customers.com/auth/customer.graphql";case"TOKEN_EXCHANGE":return `${o}/oauth/token`;case"LOGOUT":return `${o}/logout`}}}function hn(e,t){if(!e.url)return t;let{pathname:r}=new URL(e.url),o=t+`?${new URLSearchParams({return_to:r}).toString()}`;return Re(o)}function wt({session:e,customerAccountId:t,shopId:r,customerApiVersion:o=Be,request:n,waitUntil:a,authUrl:s,customAuthStatusHandler:c,logErrors:i=!0,unstableB2b:u=!1,loginPath:p="/account/login",authorizePath:y="/account/authorize",defaultRedirectPath:m="/account"}){if(o!==Be&&console.warn(`[h2:warn:createCustomerAccountClient] You are using Customer Account API version ${o} when this version of Hydrogen was built for ${Be}.`),e||console.warn("[h2:warn:createCustomerAccountClient] session is required to use Customer Account API. Ensure the session object passed in exist."),!n?.url)throw new Error("[h2:error:createCustomerAccountClient] The request object does not contain a URL.");let l=c||(()=>hn(n,p)),d=new URL(n.url),g=d.protocol==="http:"?d.origin.replace("http","https"):d.origin,P=ht({requestUrl:g,defaultUrl:y,redirectUrl:s}),f=Ur(o,r),h=Cn(f,t),A=f("GRAPHQL"),b={};async function _({query:C,type:S,variables:v={}}){let I=await K();if(!I)throw l();new Date().getTime();let N=await fetch(A,{method:"POST",headers:{"Content-Type":"application/json","User-Agent":de,Origin:g,Authorization:I},body:JSON.stringify({query:C,variables:v})});let F=await N.text(),D={url:A,response:N,type:S,query:C,queryVariables:v,errors:void 0,client:"customer"};if(!N.ok){if(N.status===401)throw ye(e),l();let O;try{O=H(F);}catch{O=[{message:F}];}Te({...D,errors:O});}try{let O=H(F),{errors:J}=O,et=J?.map(({message:De,...tt})=>new ne(De,{...tt,clientOperation:`customerAccount.${D.type}`,requestId:N.headers.get("x-request-id"),queryVariables:v,query:C}));return {...O,...J&&{errors:et}}}catch{Te({...D,errors:[{message:F}]});}}async function k(){if(!r)return !1;let C=e.get(L),S=C?.accessToken,v=C?.expiresAt;if(!S||!v)return !1;let I=$?.();try{await wr({locks:b,expiresAt:v,session:e,customerAccountId:t,customerAccountTokenExchangeUrl:f("TOKEN_EXCHANGE"),httpsOrigin:g,debugInfo:{waitUntil:a,stackInfo:I,...Y(n)},exchangeForStorefrontCustomerAccessToken:Ce});}catch{return !1}return !0}async function x(){if(!await k())throw l()}async function K(){if(await k())return e.get(L)?.accessToken}async function he(C,S){return h(),C=ue(C),Fe(C,"customer.mutate"),pe(_({query:C,type:"mutation",...S}),{logErrors:i})}async function te(C,S){return h(),C=ue(C),Ve(C,"customer.query"),pe(_({query:C,type:"query",...S}),{logErrors:i})}function z(C){e.set(le,{...e.get(le),...C});}async function ie(){if(await k())return e.get(le)}async function Ce(){if(!u)return;let C=`#graphql
      mutation storefrontCustomerAccessTokenCreate {
        storefrontCustomerAccessTokenCreate {
          customerAccessToken
        }
      }
    `,{data:S}=await he(C),v=S?.storefrontCustomerAccessTokenCreate?.customerAccessToken;v&&z({customerAccessToken:v});}return {login:async C=>{h();let S=new URL(f("AUTH")),v=Dr(),I=Ge();if(S.searchParams.set("client_id",t),S.searchParams.set("scope","openid email"),S.searchParams.append("response_type","code"),S.searchParams.append("redirect_uri",P),S.searchParams.set("scope",f("LOGIN_SCOPE")),S.searchParams.append("state",v),S.searchParams.append("nonce",I),C?.uiLocales){let[N,F]=C.uiLocales.split("-"),D=N.toLowerCase();F&&(D+=`-${F.toUpperCase()}`),S.searchParams.append("ui_locales",D);}let q=br(),re=await Or(q);return e.set(L,{...e.get(L),codeVerifier:q,state:v,nonce:I,redirectPath:qe(n.url)||B(n,"Referer")||m}),S.searchParams.append("code_challenge",re),S.searchParams.append("code_challenge_method","S256"),Re(S.toString())},logout:async C=>{h();let S=e.get(L)?.idToken,v=ht({requestUrl:g,defaultUrl:g,redirectUrl:C?.postLogoutRedirectUri}),I=S?new URL(`${f("LOGOUT")}?${new URLSearchParams([["id_token_hint",S],["post_logout_redirect_uri",v]]).toString()}`).toString():v;return ye(e),Re(I,{headers:C?.headers||{}})},isLoggedIn:k,handleAuthStatus:x,getAccessToken:K,getApiUrl:()=>A,mutate:he,query:te,authorize:async()=>{h();let C=d.searchParams.get("code"),S=d.searchParams.get("state");if(!C||!S)throw ye(e),new V("Unauthorized","No code or state parameter found in the redirect URL.");if(e.get(L)?.state!==S)throw ye(e),new V("Unauthorized","The session state does not match the state parameter. Make sure that the session is configured correctly and passed to `createCustomerAccountClient`.");let v=t,I=new URLSearchParams;I.append("grant_type","authorization_code"),I.append("client_id",v),I.append("redirect_uri",P),I.append("code",C);let q=e.get(L)?.codeVerifier;if(!q)throw new V("Unauthorized","No code verifier found in the session. Make sure that the session is configured correctly and passed to `createCustomerAccountClient`.");I.append("code_verifier",q);let re={"content-type":"application/x-www-form-urlencoded","User-Agent":de,Origin:g},N=$?.();new Date().getTime();let D=f("TOKEN_EXCHANGE"),O=await fetch(D,{method:"POST",headers:re,body:I});if(!O.ok)throw new Response(await O.text(),{status:O.status,headers:{"Content-Type":"text/html; charset=utf-8"}});let{access_token:J,expires_in:et,id_token:De,refresh_token:tt}=await O.json(),zt=e.get(L)?.nonce,Jt=await _r(De);if(zt!==Jt)throw new V("Unauthorized",`Returned nonce does not match: ${zt} !== ${Jt}`);let Yt=J;r||(Yt=await Rt(J,t,f("TOKEN_EXCHANGE"),g,{waitUntil:a,stackInfo:N,...Y(n)}));let Ao=e.get(L)?.redirectPath;return e.set(L,{accessToken:Yt,expiresAt:new Date(new Date().getTime()+(et-120)*1e3).getTime()+"",refreshToken:tt,idToken:De}),await Ce(),Re(Ao||m)},UNSTABLE_setBuyer:z,UNSTABLE_getBuyer:ie}}function Cn(e,t){return function(){try{if(!t)throw Error();new URL(e("CA_BASE_URL")),new URL(e("CA_BASE_AUTH_URL"));}catch{console.error(new Error("[h2:error:customerAccount] You do not have the valid credential to use Customer Account API.\nRun `h2 env pull` to link your store credentials."));let o="Internal Server Error";throw new Response(o,{status:500})}}}var Pn="https://raw.githubusercontent.com/Shopify/hydrogen/main/docs/changelog.json";async function An({request:e,changelogUrl:t}){new URL(e.url).searchParams;return fetch(t||Pn)}var Lr="cartFormInput";function W({children:e,action:t,inputs:r,route:o,fetcherKey:n}){let a=react$1.useFetcher({key:n});return jsxRuntime.jsxs(a.Form,{action:o||"",method:"post",children:[(t||r)&&jsxRuntime.jsx("input",{type:"hidden",name:Lr,value:JSON.stringify({action:t,inputs:r})}),typeof e=="function"?e(a):e]})}W.INPUT_NAME=Lr;W.ACTIONS={AttributesUpdateInput:"AttributesUpdateInput",BuyerIdentityUpdate:"BuyerIdentityUpdate",Create:"Create",DiscountCodesUpdate:"DiscountCodesUpdate",GiftCardCodesUpdate:"GiftCardCodesUpdate",LinesAdd:"LinesAdd",LinesRemove:"LinesRemove",LinesUpdate:"LinesUpdate",NoteUpdate:"NoteUpdate",SelectedDeliveryOptionsUpdate:"SelectedDeliveryOptionsUpdate",MetafieldsSet:"MetafieldsSet",MetafieldDelete:"MetafieldDelete"};function Tn(e){let t={};for(let s of e.entries()){let c=s[0],i=e.getAll(c);t[c]=i.length>1?i:s[1];}let{cartFormInput:r,...o}=t,{action:n,inputs:a}=r?JSON.parse(String(r)):{};return {action:n,inputs:{...a,...o}}}W.getFormInput=Tn;var E=`#graphql
  fragment CartApiError on CartUserError {
    message
    field
    code
  }
`,R=`#graphql
  fragment CartApiMutation on Cart {
    id
    totalQuantity
    checkoutUrl
  }
`,w=`#graphql
  fragment CartApiWarning on CartWarning {
    code
    message
    target
  }
`;function bt(e){return async(t,r)=>{let o=e.customerAccount?await e.customerAccount.UNSTABLE_getBuyer():void 0,{cartId:n,...a}=r||{},{buyerIdentity:s,...c}=t,{cartCreate:i,errors:u}=await e.storefront.mutate(En(e.cartFragment),{variables:{input:{...c,buyerIdentity:{...o,...s}},...a}});return T(i,u)}}var En=(e=R)=>`#graphql
  mutation cartCreate(
    $input: CartInput!
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cartCreate(input: $input) {
      cart {
        ...CartApiMutation
        checkoutUrl
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Ot({storefront:e,customerAccount:t,getCartId:r,cartFragment:o}){return async n=>{let a=r();if(!a)return null;let[s,{cart:c,errors:i}]=await Promise.all([t?t.isLoggedIn():!1,e.query(Rn(o),{variables:{cartId:a,...n},cache:e.CacheNone()})]);if(s&&c?.checkoutUrl){let u=new URL(c.checkoutUrl);u.searchParams.set("logged_in","true"),c.checkoutUrl=u.toString();}return c||i?T(c,i):null}}var Rn=(e=wn)=>`#graphql
  query CartQuery(
    $cartId: ID!
    $numCartLines: Int = 100
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cart(id: $cartId) {
      ...CartApiQuery
    }
  }

  ${e}
`,wn=`#graphql
  fragment CartApiQuery on Cart {
    updatedAt
    id
    checkoutUrl
    totalQuantity
    buyerIdentity {
      countryCode
      customer {
        id
        email
        firstName
        lastName
        displayName
      }
      email
      phone
    }
    lines(first: $numCartLines) {
      edges {
        node {
          id
          quantity
          attributes {
            key
            value
          }
          cost {
            totalAmount {
              amount
              currencyCode
            }
            amountPerQuantity {
              amount
              currencyCode
            }
            compareAtAmountPerQuantity {
              amount
              currencyCode
            }
          }
          merchandise {
            ... on ProductVariant {
              id
              availableForSale
              compareAtPrice {
                ...CartApiMoney
              }
              price {
                ...CartApiMoney
              }
              requiresShipping
              title
              image {
                ...CartApiImage
              }
              product {
                handle
                title
                id
                vendor
              }
              selectedOptions {
                name
                value
              }
            }
          }
        }
      }
    }
    cost {
      subtotalAmount {
        ...CartApiMoney
      }
      totalAmount {
        ...CartApiMoney
      }
      totalDutyAmount {
        ...CartApiMoney
      }
      totalTaxAmount {
        ...CartApiMoney
      }
    }
    note
    attributes {
      key
      value
    }
    discountCodes {
      applicable
      code
    }
  }

  fragment CartApiMoney on MoneyV2 {
    currencyCode
    amount
  }

  fragment CartApiImage on Image {
    id
    url
    altText
    width
    height
  }
`;function xt(e){return async(t,r)=>{let{cartLinesAdd:o,errors:n}=await e.storefront.mutate(bn(e.cartFragment),{variables:{cartId:e.getCartId(),lines:t,...r}});return T(o,n)}}var bn=(e=R)=>`#graphql
  mutation cartLinesAdd(
    $cartId: ID!
    $lines: [CartLineInput!]!
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cartLinesAdd(cartId: $cartId, lines: $lines) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;var kr="__h_pending_";function Nr(e){return kr+e}function Qe(e){return e.startsWith(kr)}function We(e,t){if(t.some(r=>Qe(typeof r=="string"?r:r.id)))throw new Error(`Tried to perform an action on an optimistic line. Make sure to disable your "${e}" CartForm action when the line is optimistic.`)}function Dt(e){return async(t,r)=>{We("updateLines",t);let{cartLinesUpdate:o,errors:n}=await e.storefront.mutate(On(e.cartFragment),{variables:{cartId:e.getCartId(),lines:t,...r}});return T(o,n)}}var On=(e=R)=>`#graphql
  mutation cartLinesUpdate(
    $cartId: ID!
    $lines: [CartLineUpdateInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartLinesUpdate(cartId: $cartId, lines: $lines) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function _t(e){return async(t,r)=>{We("removeLines",t);let{cartLinesRemove:o,errors:n}=await e.storefront.mutate(xn(e.cartFragment),{variables:{cartId:e.getCartId(),lineIds:t,...r}});return T(o,n)}}var xn=(e=R)=>`#graphql
  mutation cartLinesRemove(
    $cartId: ID!
    $lineIds: [ID!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartLinesRemove(cartId: $cartId, lineIds: $lineIds) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Ut(e){return async(t,r)=>{let o=t.filter((s,c,i)=>i.indexOf(s)===c),{cartDiscountCodesUpdate:n,errors:a}=await e.storefront.mutate(Dn(e.cartFragment),{variables:{cartId:e.getCartId(),discountCodes:o,...r}});return T(n,a)}}var Dn=(e=R)=>`#graphql
  mutation cartDiscountCodesUpdate(
    $cartId: ID!
    $discountCodes: [String!]
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartDiscountCodesUpdate(cartId: $cartId, discountCodes: $discountCodes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Lt(e){return async(t,r)=>{let o=t.filter((s,c,i)=>i.indexOf(s)===c),{cartGiftCardCodesUpdate:n,errors:a}=await e.storefront.mutate(_n(e.cartFragment),{variables:{cartId:e.getCartId(),giftCardCodes:o,...r}});return T(n,a)}}var _n=(e=R)=>`#graphql
  mutation cartGiftCardCodesUpdate(
    $cartId: ID!
    $giftCardCodes: [String!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartGiftCardCodesUpdate(cartId: $cartId, giftCardCodes: $giftCardCodes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function kt(e){return async(t,r)=>{t.companyLocationId&&e.customerAccount&&e.customerAccount.UNSTABLE_setBuyer({companyLocationId:t.companyLocationId});let o=e.customerAccount?await e.customerAccount.UNSTABLE_getBuyer():void 0,{cartBuyerIdentityUpdate:n,errors:a}=await e.storefront.mutate(Un(e.cartFragment),{variables:{cartId:e.getCartId(),buyerIdentity:{...o,...t},...r}});return T(n,a)}}var Un=(e=R)=>`#graphql
  mutation cartBuyerIdentityUpdate(
    $cartId: ID!
    $buyerIdentity: CartBuyerIdentityInput!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartBuyerIdentityUpdate(cartId: $cartId, buyerIdentity: $buyerIdentity) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Nt(e){return async(t,r)=>{let{cartNoteUpdate:o,errors:n}=await e.storefront.mutate(Ln(e.cartFragment),{variables:{cartId:e.getCartId(),note:t,...r}});return T(o,n)}}var Ln=(e=R)=>`#graphql
  mutation cartNoteUpdate(
    $cartId: ID!
    $note: String!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartNoteUpdate(cartId: $cartId, note: $note) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Mt(e){return async(t,r)=>{let{cartSelectedDeliveryOptionsUpdate:o,errors:n}=await e.storefront.mutate(kn(e.cartFragment),{variables:{cartId:e.getCartId(),selectedDeliveryOptions:t,...r}});return T(o,n)}}var kn=(e=R)=>`#graphql
  mutation cartSelectedDeliveryOptionsUpdate(
    $cartId: ID!
    $selectedDeliveryOptions: [CartSelectedDeliveryOptionInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartSelectedDeliveryOptionsUpdate(cartId: $cartId, selectedDeliveryOptions: $selectedDeliveryOptions) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Vt(e){return async(t,r)=>{let{cartAttributesUpdate:o,errors:n}=await e.storefront.mutate(Nn(e.cartFragment),{variables:{cartId:r?.cartId||e.getCartId(),attributes:t}});return T(o,n)}}var Nn=(e=R)=>`#graphql
  mutation cartAttributesUpdate(
    $cartId: ID!
    $attributes: [AttributeInput!]!
  ) {
    cartAttributesUpdate(cartId: $cartId, attributes: $attributes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${E}
  ${w}
`;function Ft(e){return async(t,r)=>{let o=r?.cartId||e.getCartId(),n=t.map(c=>({...c,ownerId:o})),{cartMetafieldsSet:a,errors:s}=await e.storefront.mutate(Mn(),{variables:{metafields:n}});return T({cart:{id:o},...a},s)}}var Mn=()=>`#graphql
  mutation cartMetafieldsSet(
    $metafields: [CartMetafieldsSetInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartMetafieldsSet(metafields: $metafields) {
      userErrors {
        code
        elementIndex
        field
        message
      }
    }
  }
`;function qt(e){return async(t,r)=>{let o=r?.cartId||e.getCartId(),{cartMetafieldDelete:n,errors:a}=await e.storefront.mutate(Vn(),{variables:{input:{ownerId:o,key:t}}});return T({cart:{id:o},...n},a)}}var Vn=()=>`#graphql
  mutation cartMetafieldDelete(
    $input: CartMetafieldDeleteInput!
  ) {
    cartMetafieldDelete(input: $input) {
      userErrors {
        code
        field
        message
      }
    }
  }
`;var $t=e=>{let t=cookie.parse(mt(e,"Cookie")||"");return ()=>t.cart?`gid://shopify/Cart/${t.cart}`:void 0};var Ht=e=>t=>{let r=new Headers;return r.append("Set-Cookie",cookie.stringify("cart",t.split("/").pop()||"",{path:"/",...e})),r};function Bt(e){let{getCartId:t,setCartId:r,storefront:o,customerAccount:n,cartQueryFragment:a,cartMutateFragment:s}=e,c=t(),i=()=>c||t(),u={storefront:o,getCartId:i,cartFragment:s,customerAccount:n},p=bt(u),y=async function(...l){let d=await p(...l);return c=d?.cart?.id,d},m={get:Ot({storefront:o,customerAccount:n,getCartId:i,cartFragment:a}),getCartId:i,setCartId:r,create:y,addLines:async(l,d)=>{let g=l.map(P=>({attributes:P.attributes,quantity:P.quantity,merchandiseId:P.merchandiseId,sellingPlanId:P.sellingPlanId}));return c||d?.cartId?await xt(u)(g,d):await y({lines:g},d)},updateLines:Dt(u),removeLines:_t(u),updateDiscountCodes:async(l,d)=>c||d?.cartId?await Ut(u)(l,d):await y({discountCodes:l},d),updateGiftCardCodes:async(l,d)=>c||d?.cartId?await Lt(u)(l,d):await y({giftCardCodes:l},d),updateBuyerIdentity:async(l,d)=>c||d?.cartId?await kt(u)(l,d):await y({buyerIdentity:l},d),updateNote:async(l,d)=>c||d?.cartId?await Nt(u)(l,d):await y({note:l},d),updateSelectedDeliveryOption:Mt(u),updateAttributes:async(l,d)=>c||d?.cartId?await Vt(u)(l,d):await y({attributes:l},d),setMetafields:async(l,d)=>c||d?.cartId?await Ft(u)(l,d):await y({metafields:l},d),deleteMetafield:qt(u)};return "customMethods"in e?{...m,...e.customMethods??{}}:m}function Hn(e){let t=react$1.useFetchers();if(!t||!t.length)return e;let r=e?.lines?structuredClone(e):{lines:{nodes:[]}},o=r.lines.nodes,n=!1;for(let{formData:a}of t){if(!a)continue;let s=W.getFormInput(a);if(s.action===W.ACTIONS.LinesAdd)for(let c of s.inputs.lines){if(!c.selectedVariant){console.error("[h2:error:useOptimisticCart] No selected variant was passed in the cart action. Make sure to pass the selected variant if you want to use an optimistic cart");continue}let i=o.find(u=>u.merchandise.id===c.selectedVariant?.id);n=!0,i?(i.quantity=(i.quantity||1)+(c.quantity||1),i.isOptimistic=!0):o.unshift({id:Nr(c.selectedVariant.id),merchandise:c.selectedVariant,isOptimistic:!0,quantity:c.quantity||1});}else if(s.action===W.ACTIONS.LinesRemove)for(let c of s.inputs.lineIds){let i=o.findIndex(u=>u.id===c);if(i!==-1){if(Qe(o[i].id)){console.error("[h2:error:useOptimisticCart] Tried to remove an optimistic line that has not been added to the cart yet");continue}o.splice(i,1),n=!0;}else console.warn(`[h2:warn:useOptimisticCart] Tried to remove line '${c}' but it doesn't exist in the cart`);}else if(s.action===W.ACTIONS.LinesUpdate)for(let c of s.inputs.lines){let i=o.findIndex(u=>c.id===u.id);if(i>-1){if(Qe(o[i].id)){console.error("[h2:error:useOptimisticCart] Tried to update an optimistic line that has not been added to the cart yet");continue}o[i].quantity=c.quantity,o[i].quantity===0&&o.splice(i,1),n=!0;}else console.warn(`[h2:warn:useOptimisticCart] Tried to update line '${c.id}' but it doesn't exist in the cart`);}}return n&&(r.isOptimistic=n),r.totalQuantity=o.reduce((a,s)=>a+s.quantity,0),r}function Kn({handle:e,options:t=[],variants:r=[],productPath:o="products",waitForNavigation:n=!1,selectedVariant:a,children:s}){let c=t;c[0]?.values&&(Me("[h2:warn:VariantSelector] product.options.values is deprecated. Use product.options.optionValues instead."),c[0]&&!c[0].optionValues&&(c=t.map(d=>({...d,optionValues:d.values?.map(g=>({name:g}))||[]}))));let i=r instanceof Array?r:hydrogenReact.flattenConnection(r),{searchParams:u,path:p,alreadyOnProductPage:y}=Jn(e,o,n),m=c.filter(d=>d?.optionValues?.length===1),l=a?a?.selectedOptions?.reduce((d,g)=>(d[g.name]=g.value,d),{}):{};return react.createElement(react.Fragment,null,...react.useMemo(()=>c.map(d=>{let g,P=[];for(let f of d.optionValues){let h=new URLSearchParams(y?u:void 0);h.set(d.name,f.name),m.forEach(x=>{x.optionValues[0].name&&h.set(x.name,x.optionValues[0].name);});let A=i.find(x=>x?.selectedOptions?.every(K=>(h.get(K?.name)||l?.[K?.name])===K?.value)),b=u.get(d.name);!b&&a&&(b=l?.[d.name]||null);let _=b?b===f.name:!1;_&&(g=f.name);let k="?"+h.toString();P.push({value:f.name,optionValue:f,isAvailable:A?A.availableForSale:!0,to:p+k,search:k,isActive:_,variant:A});}return s({option:{name:d.name,value:g,values:P}})}),[c,i,s]))}var zn=e=>{if(typeof e?.url>"u")throw new TypeError(`Expected a Request instance, got ${typeof e}`);let t=new URL(e.url).searchParams,r=[];return t.forEach((o,n)=>{r.push({name:n,value:o});}),r};function Jn(e,t,r){let{pathname:o,search:n}=react$1.useLocation(),a=react$1.useNavigation();return react.useMemo(()=>{let s=/(\/[a-zA-Z]{2}-[a-zA-Z]{2}\/)/g.exec(o),c=s&&s.length>0;t=t.startsWith("/")?t.substring(1):t;let i=c?`${s[0]}${t}/${e}`:`/${t}/${e}`;return {searchParams:new URLSearchParams(r||a.state!=="loading"?n:a.location.search),alreadyOnProductPage:i===o,path:i}},[o,n,r,e,t,a])}function ea(e,t){let r=react$1.useNavigation(),[o,n]=react.useState([]);if(react.useEffect(()=>{Promise.resolve(t).then(a=>{a&&n(a instanceof Array?a:a.product?.variants?.nodes||[]);}).catch(a=>{reportError(new Error("[h2:error:useOptimisticVariant] An error occurred while resolving the variants for the optimistic product hook.",{cause:a}));});},[JSON.stringify(t)]),r.state==="loading"){let a=new URLSearchParams(r.location.search),s=!1,c=o.find(i=>i.selectedOptions?i.selectedOptions.every(u=>a.get(u.name)===u.value):(s||(s=!0,reportError(new Error("[h2:error:useOptimisticVariant] The optimistic product hook requires your product query to include variants with the selectedOptions field."))),!1));if(c)return {...c,isOptimistic:!0}}return e}var Vr=react.createContext(void 0),aa=Vr.Provider,Gt=()=>react.useContext(Vr);function sa(e){let t=Ge(),r=ia(t,e);return {nonce:t,header:r,NonceProvider:({children:n})=>react.createElement(aa,{value:t},n)}}function ia(e,t){let{shop:r,...o}=t??{},n=`'nonce-${e}'`,a=["'self'","'unsafe-inline'","https://cdn.shopify.com"],s=["'self'","https://monorail-edge.shopifysvc.com"];r&&r.checkoutDomain&&s.push(`https://${r.checkoutDomain}`),r&&r.storeDomain&&s.push(`https://${r.storeDomain}`);let i={baseUri:["'self'"],defaultSrc:["'self'",n,"https://cdn.shopify.com","https://shopify.com"],frameAncestors:["'none'"],styleSrc:a,connectSrc:s},u=Object.assign({},i,o);for(let p in i){let y=o[p];p&&y&&(u[p]=ca(y,i[p]));}return u.scriptSrc instanceof Array?u.scriptSrc=[...u.scriptSrc.filter(p=>!p.startsWith("'nonce")),n]:u.defaultSrc instanceof Array&&(u.defaultSrc=[...u.defaultSrc.filter(p=>!p.startsWith("'nonce")),n]),na__default.default({directives:u})}function ca(e,t){let r=typeof t=="string"?[t]:t,o=Array.isArray(e)?e:[String(e)];return Array.isArray(r)?r.every(a=>a==="'none'")?o:[...o,...r]:r}var da=react.forwardRef((e,t)=>{let{waitForHydration:r,src:o,...n}=e;if(r)return jsxRuntime.jsx(la,{src:o,options:n});let a=Gt();return jsxRuntime.jsx("script",{suppressHydrationWarning:!0,...n,src:o,nonce:a,ref:t})});function la({src:e,options:t}){if(!e)throw new Error("`waitForHydration` with the Script component requires a `src` prop");return hydrogenReact.useLoadScript(e,{attributes:t}),null}function fa(e){let t=react$1.useFetchers(),r={};for(let{formData:o}of t)if(o?.get("optimistic-identifier")===e)try{if(o.has("optimistic-data")){let n=JSON.parse(String(o.get("optimistic-data")));Object.assign(r,n);}}catch{}return r}function ma({id:e,data:t}){return jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsx("input",{type:"hidden",name:"optimistic-identifier",value:e}),jsxRuntime.jsx("input",{type:"hidden",name:"optimistic-data",value:JSON.stringify(t)})]})}function Pa(e){return jsxRuntime.jsx(hydrogenReact.ShopPayButton,{channel:"hydrogen",...e})}function fe(e){let{type:t,data:r={},customData:o}=e,n=react$1.useLocation(),{publish:a,cart:s,prevCart:c,shop:i,customData:u}=j(),p=n.pathname+n.search,y={...r,customData:{...u,...o},cart:s,prevCart:c,shop:i};return react.useEffect(()=>{i?.shopId&&(y={...y,url:window.location.href},a(t,y));},[a,p,i?.shopId]),null}function $r(e){return jsxRuntime.jsx(fe,{...e,type:"page_viewed"})}function Hr(e){return jsxRuntime.jsx(fe,{...e,type:"product_viewed"})}function Br(e){return jsxRuntime.jsx(fe,{...e,type:"collection_viewed"})}function Gr(e){return jsxRuntime.jsx(fe,{...e,type:"cart_viewed"})}function Qr(e){return jsxRuntime.jsx(fe,{...e,type:"search_viewed"})}function Wr(e){return jsxRuntime.jsx(fe,{...e})}var M={PAGE_VIEWED:"page_viewed",PRODUCT_VIEWED:"product_viewed",COLLECTION_VIEWED:"collection_viewed",CART_VIEWED:"cart_viewed",SEARCH_VIEWED:"search_viewed",CART_UPDATED:"cart_updated",PRODUCT_ADD_TO_CART:"product_added_to_cart",PRODUCT_REMOVED_FROM_CART:"product_removed_from_cart",CUSTOM_EVENT:"custom_"};var wa="https://cdn.shopify.com/shopifycloud/consent-tracking-api/v0.1/consent-tracking-api.js",ba="https://cdn.shopify.com/shopifycloud/privacy-banner/storefront-banner.js";function Qt(e){console.error(`[h2:error:useCustomerPrivacy] Unable to setup Customer Privacy API: Missing consent.${e} configuration.`);}function Wt(e){let{withPrivacyBanner:t=!1,onVisitorConsentCollected:r,onReady:o,...n}=e;hydrogenReact.useLoadScript(t?ba:wa,{attributes:{id:"customer-privacy-api"}});let{observing:a,setLoaded:s}=Oa({withPrivacyBanner:t,onLoaded:o}),c=react.useMemo(()=>{let{checkoutDomain:u,storefrontAccessToken:p}=n;return u||Qt("checkoutDomain"),p||Qt("storefrontAccessToken"),(p.startsWith("shpat_")||p.length!==32)&&console.error("[h2:error:useCustomerPrivacy] It looks like you passed a private access token, make sure to use the public token"),{checkoutRootDomain:u,storefrontAccessToken:p,storefrontRootDomain:zr(u),country:n.country,locale:n.locale}},[n,zr,Qt]);react.useEffect(()=>{let u=p=>{r&&r(p.detail);};return document.addEventListener("visitorConsentCollected",u),()=>{document.removeEventListener("visitorConsentCollected",u);}},[r]),react.useEffect(()=>{if(!t||a.current.privacyBanner)return;a.current.privacyBanner=!0;let u=window.privacyBanner||void 0;Object.defineProperty(window,"privacyBanner",{configurable:!0,get(){return u},set(y){if(typeof y=="object"&&y!==null&&"showPreferences"in y&&"loadBanner"in y){let m=y;m.loadBanner(c),u=Yr({privacyBanner:m,config:c}),s.privacyBanner(),Kr();}}});},[t,c,Yr,s.privacyBanner]),react.useEffect(()=>{if(a.current.customerPrivacy)return;a.current.customerPrivacy=!0;let u=null,p=window.Shopify||void 0;Object.defineProperty(window,"Shopify",{configurable:!0,get(){return p},set(y){typeof y=="object"&&y!==null&&Object.keys(y).length===0&&(p=y,Object.defineProperty(window.Shopify,"customerPrivacy",{configurable:!0,get(){return u},set(m){if(typeof m=="object"&&m!==null&&"setTrackingConsent"in m){let l=m;u={...l,setTrackingConsent:Jr({customerPrivacy:l,config:c})},p={...p,customerPrivacy:u},s.customerPrivacy(),Kr();}}}));}});},[c,Jr,s.customerPrivacy]);let i={customerPrivacy:ge()};return t&&(i.privacyBanner=Ke()),i}var jr=!1;function Kr(){if(jr)return;jr=!0;let e=new CustomEvent("shopifyCustomerPrivacyApiLoaded");document.dispatchEvent(e);}function Oa({withPrivacyBanner:e,onLoaded:t}){let r=react.useRef({customerPrivacy:!1,privacyBanner:!1}),[o,n]=react.useState(e?[!1,!1]:[!1]),a=o.every(Boolean),s={customerPrivacy:()=>{n(e?c=>[!0,c[1]]:()=>[!0]);},privacyBanner:()=>{e&&n(c=>[c[0],!0]);}};return react.useEffect(()=>{a&&t&&t();},[a,t]),{observing:r,setLoaded:s}}function zr(e){if(typeof window>"u")return;let t=window.document.location.host,r=e.split(".").reverse(),o=t.split(".").reverse(),n=[];return r.forEach((a,s)=>{a===o[s]&&n.push(a);}),n.reverse().join(".")}function Jr({customerPrivacy:e,config:t}){let r=e.setTrackingConsent,{locale:o,country:n,...a}=t;function s(c,i){r({...a,headlessStorefront:!0,...c},i);}return s}function Yr({privacyBanner:e,config:t}){let r=e.loadBanner,o=e.showPreferences;function n(s){if(typeof s=="object"){r({...t,...s});return}r(t);}function a(s){if(typeof s=="object"){o({...t,...s});return}o(t);}return {loadBanner:n,showPreferences:a}}function ge(){try{return window.Shopify&&window.Shopify.customerPrivacy?window.Shopify?.customerPrivacy:null}catch{return null}}function Ke(){try{return window&&window?.privacyBanner?window.privacyBanner:null}catch{return null}}var Xr="2025.1.0";function La(){let e=ge();if(!e)throw new Error("Shopify Customer Privacy API not available. Must be used within a useEffect. Make sure to load the Shopify Customer Privacy API with useCustomerPrivacy() or <AnalyticsProvider>.");return e}function to({consent:e,onReady:t,domain:r}){let{subscribe:o,register:n,canTrack:a}=j(),[s,c]=react.useState(!1),[i,u]=react.useState(!1),p=react.useRef(!1),{checkoutDomain:y,storefrontAccessToken:m,language:l}=e,{ready:d}=n("Internal_Shopify_Analytics");return Wt({...e,locale:l,checkoutDomain:y||"mock.shop",storefrontAccessToken:m||"abcdefghijklmnopqrstuvwxyz123456",onVisitorConsentCollected:()=>u(!0),onReady:()=>u(!0)}),hydrogenReact.useShopifyCookies({hasUserConsent:i?a():!0,domain:r,checkoutDomain:y}),react.useEffect(()=>{p.current||(p.current=!0,o(M.PAGE_VIEWED,Na),o(M.PRODUCT_VIEWED,Ma),o(M.COLLECTION_VIEWED,Va),o(M.SEARCH_VIEWED,Fa),o(M.PRODUCT_ADD_TO_CART,qa),c(!0));},[o]),react.useEffect(()=>{s&&i&&(d(),t());},[s,i,t]),null}function ze(e){console.error(`[h2:error:ShopifyAnalytics] Unable to send Shopify analytics: Missing shop.${e} configuration.`);}function Oe(e){let t=La(),r=t.analyticsProcessingAllowed();if(!e?.shop?.shopId){ze("shopId");return}if(!e?.shop?.acceptedLanguage){ze("acceptedLanguage");return}if(!e?.shop?.currency){ze("currency");return}if(!e?.shop?.hydrogenSubchannelId){ze("hydrogenSubchannelId");return}return {shopifySalesChannel:"hydrogen",assetVersionId:Xr,...e.shop,hasUserConsent:r,...hydrogenReact.getClientBrowserParameters(),ccpaEnforced:!t.saleOfDataAllowed(),gdprEnforced:!(t.marketingAllowed()&&t.analyticsProcessingAllowed()),analyticsAllowed:t.analyticsProcessingAllowed(),marketingAllowed:t.marketingAllowed(),saleOfDataAllowed:t.saleOfDataAllowed()}}function ka(e,t){if(t===null)return;let r=Oe(e);return r?{...r,cartId:t.id}:void 0}var ee={};function Na(e){let t=Oe(e);t&&(hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.PAGE_VIEW_2,payload:{...t,...ee}}),ee={});}function Ma(e){let t=Oe(e);if(t&&ro({type:"product",products:e.products})){let r=jt(e.products);ee={pageType:hydrogenReact.AnalyticsPageType.product,resourceId:r[0].productGid},t={...t,...ee,products:jt(e.products)},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.PRODUCT_VIEW,payload:t});}}function Va(e){let t=Oe(e);t&&(ee={pageType:hydrogenReact.AnalyticsPageType.collection,resourceId:e.collection.id},t={...t,...ee,collectionHandle:e.collection.handle,collectionId:e.collection.id},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.COLLECTION_VIEW,payload:t}));}function Fa(e){let t=Oe(e);t&&(ee={pageType:hydrogenReact.AnalyticsPageType.search},t={...t,...ee,searchString:e.searchTerm},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.SEARCH_VIEW,payload:t}));}function qa(e){let{cart:t,currentLine:r}=e,o=ka(e,t);!o||!r?.id||$a({matchedLine:r,eventPayload:o});}function $a({matchedLine:e,eventPayload:t}){let r={id:e.merchandise.product.id,variantId:e.merchandise.id,title:e.merchandise.product.title,variantTitle:e.merchandise.title,vendor:e.merchandise.product.vendor,price:e.merchandise.price.amount,quantity:e.quantity,productType:e.merchandise.product.productType,sku:e.merchandise.sku};ro({type:"cart",products:[r]})&&hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.ADD_TO_CART,payload:{...t,products:jt([r])}});}function se(e,t,r,o){if(e==="cart"){let n=`${r?"merchandise":"merchandise.product"}.${t}`;console.error(`[h2:error:ShopifyAnalytics] Can't set up cart analytics events because the \`cart.lines[].${n}\` value is missing from your GraphQL cart query. In your project, search for where \`fragment CartLine on CartLine\` is defined and make sure \`${n}\` is part of your cart query. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/lib/fragments.ts#L25-L56.`);}else {let n=`${o||t}`;console.error(`[h2:error:ShopifyAnalytics] Can't set up product view analytics events because the \`${n}\` is missing from your \`<Analytics.ProductView>\`. Make sure \`${n}\` is part of your products data prop. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/routes/products.%24handle.tsx#L159-L165.`);}}function ro({type:e,products:t}){return !t||t.length===0?(se(e,"",!1,"data.products"),!1):(t.forEach(r=>{if(!r.id)return se(e,"id",!1),!1;if(!r.title)return se(e,"title",!1),!1;if(!r.price)return se(e,"price.amount",!0,"price"),!1;if(!r.vendor)return se(e,"vendor",!1),!1;if(!r.variantId)return se(e,"id",!0,"variantId"),!1;if(!r.variantTitle)return se(e,"title",!0,"variantTitle"),!1}),!0)}function jt(e){return e.map(t=>{let r={productGid:t.id,variantGid:t.variantId,name:t.title,variantName:t.variantTitle,brand:t.vendor,price:t.price,quantity:t.quantity||1,category:t.productType};return t.sku&&(r.sku=t.sku),t.productType&&(r.category=t.productType),r})}function ao(e){console.error(`[h2:error:CartAnalytics] Can't set up cart analytics events because the \`cart.${e}\` value is missing from your GraphQL cart query. In your project, search for where \`fragment CartApiQuery on Cart\` is defined and make sure \`${e}\` is part of your cart query. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/lib/fragments.ts#L59.`);}function so({cart:e,setCarts:t}){let{publish:r,shop:o,customData:n,canTrack:a,cart:s,prevCart:c}=j(),i=react.useRef(null);return react.useEffect(()=>{if(e)return Promise.resolve(e).then(u=>{if(u&&u.lines){if(!u.id){ao("id");return}if(!u.updatedAt){ao("updatedAt");return}}t(({cart:p,prevCart:y})=>u?.updatedAt!==p?.updatedAt?{cart:u,prevCart:p}:{cart:p,prevCart:y});}),()=>{}},[t,e]),react.useEffect(()=>{if(!s||!s?.updatedAt||s?.updatedAt===c?.updatedAt)return;let u;try{u=JSON.parse(localStorage.getItem("cartLastUpdatedAt")||"");}catch{u=null;}if(s.id===u?.id&&s.updatedAt===u?.updatedAt)return;let p={eventTimestamp:Date.now(),cart:s,prevCart:c,shop:o,customData:n};if(s.updatedAt===i.current)return;i.current=s.updatedAt,r("cart_updated",p),localStorage.setItem("cartLastUpdatedAt",JSON.stringify({id:s.id,updatedAt:s.updatedAt}));let y=c?.lines?hydrogenReact.flattenConnection(c?.lines):[],m=s.lines?hydrogenReact.flattenConnection(s.lines):[];y?.forEach(l=>{let d=m.filter(g=>l.id===g.id);if(d?.length===1){let g=d[0];l.quantity<g.quantity?r("product_added_to_cart",{...p,prevLine:l,currentLine:g}):l.quantity>g.quantity&&r("product_removed_from_cart",{...p,prevLine:l,currentLine:g});}else r("product_removed_from_cart",{...p,prevLine:l});}),m?.forEach(l=>{let d=y.filter(g=>l.id===g.id);(!d||d.length===0)&&r("product_added_to_cart",{...p,currentLine:l});});},[s,c,r,o,n,a]),null}var ja="https://cdn.shopify.com/shopifycloud/perf-kit/shopify-perf-kit-1.0.1.min.js";function io({shop:e}){let t=react.useRef(!1),{subscribe:r,register:o}=j(),{ready:n}=o("Internal_Shopify_Perf_Kit"),a=hydrogenReact.useLoadScript(ja,{attributes:{id:"perfkit","data-application":"hydrogen","data-shop-id":hydrogenReact.parseGid(e.shopId).id.toString(),"data-storefront-id":e.hydrogenSubchannelId,"data-monorail-region":"global","data-spa-mode":"true","data-resource-timing-sampling-rate":"100"}});return react.useEffect(()=>{a!=="done"||t.current||(t.current=!0,r(M.PAGE_VIEWED,()=>{window.PerfKit?.navigate();}),r(M.PRODUCT_VIEWED,()=>{window.PerfKit?.setPageType("product");}),r(M.COLLECTION_VIEWED,()=>{window.PerfKit?.setPageType("collection");}),r(M.SEARCH_VIEWED,()=>{window.PerfKit?.setPageType("search");}),r(M.CART_VIEWED,()=>{window.PerfKit?.setPageType("cart");}),n());},[r,n,a]),null}var Za={canTrack:()=>!1,cart:null,customData:{},prevCart:null,publish:()=>{},shop:null,subscribe:()=>{},register:()=>({ready:()=>{}}),customerPrivacy:null,privacyBanner:null},fo=react.createContext(Za),Xe=new Map,xe={};function mo(){return Object.values(xe).every(Boolean)}function co(e,t){Xe.has(e)||Xe.set(e,new Map),Xe.get(e)?.set(t.toString(),t);}var Ze=new Map;function uo(e,t){if(!mo()){Ze.set(e,t);return}go(e,t);}function go(e,t){(Xe.get(e)??new Map).forEach((r,o)=>{try{r(t);}catch(n){typeof n=="object"&&n instanceof Error?console.error("Analytics publish error",n.message,o,n.stack):console.error("Analytics publish error",n,o);}});}function po(e){return xe.hasOwnProperty(e)||(xe[e]=!1),{ready:()=>{xe[e]=!0,mo()&&Ze.size>0&&(Ze.forEach((t,r)=>{go(r,t);}),Ze.clear());}}}function lo(){try{return window.Shopify.customerPrivacy.analyticsProcessingAllowed()}catch{}return !1}function yo(e,t){return `[h2:error:Analytics.Provider] - ${e} is required. Make sure ${t} is defined in your environment variables. See https://h2o.fyi/analytics/consent to learn how to setup environment variables in the Shopify admin.`}function es({canTrack:e,cart:t,children:r,consent:o,customData:n={},shop:a=null,cookieDomain:s}){let c=react.useRef(!1),{shop:i}=ts(a),[u,p]=react.useState(!!e),[y,m]=react.useState({cart:null,prevCart:null}),[l,d]=react.useState(e?()=>e:()=>lo);if(i)if(/\/68817551382$/.test(i.shopId))Me("[h2:error:Analytics.Provider] - Mock shop is used. Analytics will not work properly.");else {if(!o.checkoutDomain){let P=yo("consent.checkoutDomain","PUBLIC_CHECKOUT_DOMAIN");ft(P);}if(!o.storefrontAccessToken){let P=yo("consent.storefrontAccessToken","PUBLIC_STOREFRONT_API_TOKEN");ft(P);}o?.country||(o.country="US"),o?.language||(o.language="EN"),o.withPrivacyBanner===void 0&&(o.withPrivacyBanner=!1);}let g=react.useMemo(()=>({canTrack:l,...y,customData:n,publish:l()?uo:()=>{},shop:i,subscribe:co,register:po,customerPrivacy:ge(),privacyBanner:Ke()}),[u,l,y,y.cart?.updatedAt,y.prevCart,uo,co,n,i,po,JSON.stringify(xe),ge,Ke]);return jsxRuntime.jsxs(fo.Provider,{value:g,children:[r,!!i&&jsxRuntime.jsx($r,{}),!!i&&!!t&&jsxRuntime.jsx(so,{cart:t,setCarts:m}),!!i&&o.checkoutDomain&&jsxRuntime.jsx(to,{consent:o,onReady:()=>{c.current=!0,p(!0),d(e?()=>e:()=>lo);},domain:s}),!!i&&jsxRuntime.jsx(io,{shop:i})]})}function j(){let e=react.useContext(fo);if(!e)throw new Error("[h2:error:useAnalytics] 'useAnalytics()' must be a descendent of <AnalyticsProvider/>");return e}function ts(e){let[t,r]=react.useState(null);return react.useEffect(()=>(Promise.resolve(e).then(r),()=>{}),[r,e]),{shop:t}}async function rs({storefront:e,publicStorefrontId:t="0"}){return e.query(os,{cache:e.CacheLong()}).then(({shop:r,localization:o})=>({shopId:r.id,acceptedLanguage:o.language.isoCode,currency:o.country.currency.isoCode,hydrogenSubchannelId:t}))}var os=`#graphql
  query ShopData(
    $country: CountryCode
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    shop {
      id
    }
    localization {
      country {
        currency {
          isoCode
        }
      }
      language {
        isoCode
      }
    }
  }
`,ns={CartView:Gr,CollectionView:Br,CustomView:Wr,ProductView:Hr,Provider:es,SearchView:Qr};var cs=function(e){return jsxRuntime.jsx(hydrogenReact.RichText,{...e,components:{link:({node:t})=>jsxRuntime.jsx(react$1.Link,{to:t.url,title:t.title,target:t.target,prefetch:"intent",children:t.children}),...e.components}})};function us(e){let{env:t,request:r,cache:o,waitUntil:n,i18n:a,session:s,logErrors:c,storefront:i={},customerAccount:u,cart:p={}}=e;s||console.warn("[h2:warn:createHydrogenContext] A session object is required to create hydrogen context.");let{storefront:y}=pr({cache:o,waitUntil:n,i18n:a,logErrors:c,storefrontHeaders:i.headers||ps(r),storefrontApiVersion:i.apiVersion,storefrontId:t.PUBLIC_STOREFRONT_ID,storeDomain:t.PUBLIC_STORE_DOMAIN,privateStorefrontToken:t.PRIVATE_STOREFRONT_API_TOKEN,publicStorefrontToken:t.PUBLIC_STOREFRONT_API_TOKEN}),m=wt({session:s,request:r,waitUntil:n,logErrors:c,customerApiVersion:u?.apiVersion,authUrl:u?.authUrl,customAuthStatusHandler:u?.customAuthStatusHandler,unstableB2b:u?.unstableB2b,customerAccountId:t.PUBLIC_CUSTOMER_ACCOUNT_API_CLIENT_ID,shopId:t.SHOP_ID}),l=Bt({getCartId:p.getId||$t(r.headers),setCartId:p.setId||Ht(),cartQueryFragment:p.queryFragment,cartMutateFragment:p.mutateFragment,customMethods:p.customMethods,storefront:y,customerAccount:m});return {storefront:y,customerAccount:m,cart:l,env:t,waitUntil:n,session:s}}function ps(e){return {requestGroupId:B(e,"request-id"),buyerIp:B(e,"oxygen-buyer-ip"),cookie:B(e,"cookie"),purpose:B(e,"purpose")}}var ds=`<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
`,ls=`
</sitemapindex>`,Co=`<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">`,Po="</urlset>";async function ys(e){let{storefront:t,request:r,types:o=["products","pages","collections","metaObjects","articles","blogs"],customChildSitemaps:n=[]}=e;if(!r||!r.url)throw new Error("A request object is required to generate a sitemap index");if(!t||!t.query)throw new Error("A storefront client is required to generate a sitemap index");let a=await t.query(Is);if(!a)throw console.warn("[h2:sitemap:warning] Sitemap index is available in API version 2024-10 and later"),new Response("Sitemap index not found.",{status:404});let s=new URL(r.url).origin,c=ds+o.map(i=>{if(!a[i])throw new Error(`[h2:sitemap:error] No data found for type ${i}. Check types passed to \`getSitemapIndex\``);return ms(i,a[i].pagesCount.count,s)}).join(`
`)+n.map(i=>"  <sitemap><loc>"+(s+(i.startsWith("/")?i:"/"+i))+"</loc></sitemap>").join(`
`)+ls;return new Response(c,{headers:{"Content-Type":"application/xml","Cache-Control":`max-age=${60*60*24}`}})}async function fs(e){let{storefront:t,request:r,params:o,getLink:n,locales:a=[],getChangeFreq:s,noItemsFallback:c="/"}=e;if(!o)throw new Error("[h2:sitemap:error] Remix params object is required to generate a sitemap");if(!r||!r.url)throw new Error("A request object is required to generate a sitemap");if(!t||!t.query)throw new Error("A storefront client is required to generate a index");if(!n)throw new Error("A `getLink` function to generate each resource is required to build a sitemap");if(!o.type||!o.page)throw new Response("No data found",{status:404});let i=o.type,u=Es[i];if(!u)throw new Response("Not found",{status:404});let p=await t.query(u,{variables:{page:parseInt(o.page,10)}});if(!p)throw console.warn("[h2:sitemap:warning] Sitemap is available in API version 2024-10 and later"),new Response("Sitemap not found.",{status:404});let y=new URL(r.url).origin,m="";return p?.sitemap?.resources?.items?.length?m=Co+p.sitemap.resources.items.map(l=>gs({getChangeFreq:s,url:n({type:l.type??i,baseUrl:y,handle:l.handle}),type:i,getLink:n,updatedAt:l.updatedAt,handle:l.handle,metaobjectType:l.type,locales:a,baseUrl:y})).join(`
`)+Po:m=Co+`
  <url><loc>${y+c}</loc></url>
`+Po,new Response(m,{headers:{"Content-Type":"application/xml","Cache-Control":`max-age=${60*60*24}`}})}function ms(e,t,r){let o="";for(let n=1;n<=t;n++)o+=`  <sitemap><loc>${r}/sitemap/${e}/${n}.xml</loc></sitemap>
`;return o}function gs({url:e,updatedAt:t,locales:r,type:o,getLink:n,baseUrl:a,handle:s,getChangeFreq:c,metaobjectType:i}){return `<url>
  <loc>${e}</loc>
  <lastmod>${t}</lastmod>
  <changefreq>${c?c({type:i??o,handle:s}):"weekly"}</changefreq>
${r.map(u=>hs(n({type:i??o,baseUrl:a,handle:s,locale:u}),u)).join(`
`)}
</url>
  `.trim()}function hs(e,t){return `  <xhtml:link rel="alternate" hreflang="${t}" href="${e}" />`}var Cs=`#graphql
    query SitemapProducts($page: Int!) {
      sitemap(type: PRODUCT) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Ps=`#graphql
    query SitemapCollections($page: Int!) {
      sitemap(type: COLLECTION) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,As=`#graphql
    query SitemapArticles($page: Int!) {
      sitemap(type: ARTICLE) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Ss=`#graphql
    query SitemapPages($page: Int!) {
      sitemap(type: PAGE) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Ts=`#graphql
    query SitemapBlogs($page: Int!) {
      sitemap(type: BLOG) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,vs=`#graphql
    query SitemapMetaobjects($page: Int!) {
      sitemap(type: METAOBJECT) {
        resources(page: $page) {
          items {
            handle
            updatedAt
            ... on SitemapResourceMetaobject {
              type
            }
          }
        }
      }
    }
`,Is=`#graphql
query SitemapIndex {
  products: sitemap(type: PRODUCT) {
    pagesCount {
      count
    }
  }
  collections: sitemap(type: COLLECTION) {
    pagesCount {
      count
    }
  }
  articles: sitemap(type: ARTICLE) {
    pagesCount {
      count
    }
  }
  pages: sitemap(type: PAGE) {
    pagesCount {
      count
    }
  }
  blogs: sitemap(type: BLOG) {
    pagesCount {
      count
    }
  }
  metaObjects: sitemap(type: METAOBJECT) {
    pagesCount {
      count
    }
  }
}
`,Es={products:Cs,articles:As,collections:Ps,pages:Ss,blogs:Ts,metaObjects:vs};//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartCreate
//! @see https://shopify.dev/docs/api/storefront/latest/queries/cart
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesAdd
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesUpdate
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesRemove
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartDiscountCodesUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartGiftCardCodesUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartBuyerIdentityUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartNoteUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartSelectedDeliveryOptionsUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartMetafieldsSet
//! @see https://shopify.dev/docs/api/storefront/2025-01/mutations/cartMetafieldDelete

Object.defineProperty(exports, "AnalyticsEventName", {
  enumerable: true,
  get: function () { return hydrogenReact.AnalyticsEventName; }
});
Object.defineProperty(exports, "AnalyticsPageType", {
  enumerable: true,
  get: function () { return hydrogenReact.AnalyticsPageType; }
});
Object.defineProperty(exports, "ExternalVideo", {
  enumerable: true,
  get: function () { return hydrogenReact.ExternalVideo; }
});
Object.defineProperty(exports, "IMAGE_FRAGMENT", {
  enumerable: true,
  get: function () { return hydrogenReact.IMAGE_FRAGMENT; }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function () { return hydrogenReact.Image; }
});
Object.defineProperty(exports, "MediaFile", {
  enumerable: true,
  get: function () { return hydrogenReact.MediaFile; }
});
Object.defineProperty(exports, "ModelViewer", {
  enumerable: true,
  get: function () { return hydrogenReact.ModelViewer; }
});
Object.defineProperty(exports, "Money", {
  enumerable: true,
  get: function () { return hydrogenReact.Money; }
});
Object.defineProperty(exports, "ShopifySalesChannel", {
  enumerable: true,
  get: function () { return hydrogenReact.ShopifySalesChannel; }
});
Object.defineProperty(exports, "Video", {
  enumerable: true,
  get: function () { return hydrogenReact.Video; }
});
Object.defineProperty(exports, "customerAccountApiCustomScalars", {
  enumerable: true,
  get: function () { return hydrogenReact.customerAccountApiCustomScalars; }
});
Object.defineProperty(exports, "decodeEncodedVariant", {
  enumerable: true,
  get: function () { return hydrogenReact.decodeEncodedVariant; }
});
Object.defineProperty(exports, "flattenConnection", {
  enumerable: true,
  get: function () { return hydrogenReact.flattenConnection; }
});
Object.defineProperty(exports, "getAdjacentAndFirstAvailableVariants", {
  enumerable: true,
  get: function () { return hydrogenReact.getAdjacentAndFirstAvailableVariants; }
});
Object.defineProperty(exports, "getClientBrowserParameters", {
  enumerable: true,
  get: function () { return hydrogenReact.getClientBrowserParameters; }
});
Object.defineProperty(exports, "getProductOptions", {
  enumerable: true,
  get: function () { return hydrogenReact.getProductOptions; }
});
Object.defineProperty(exports, "getShopifyCookies", {
  enumerable: true,
  get: function () { return hydrogenReact.getShopifyCookies; }
});
Object.defineProperty(exports, "isOptionValueCombinationInEncodedVariant", {
  enumerable: true,
  get: function () { return hydrogenReact.isOptionValueCombinationInEncodedVariant; }
});
Object.defineProperty(exports, "mapSelectedProductOptionToObject", {
  enumerable: true,
  get: function () { return hydrogenReact.mapSelectedProductOptionToObject; }
});
Object.defineProperty(exports, "parseGid", {
  enumerable: true,
  get: function () { return hydrogenReact.parseGid; }
});
Object.defineProperty(exports, "parseMetafield", {
  enumerable: true,
  get: function () { return hydrogenReact.parseMetafield; }
});
Object.defineProperty(exports, "sendShopifyAnalytics", {
  enumerable: true,
  get: function () { return hydrogenReact.sendShopifyAnalytics; }
});
Object.defineProperty(exports, "storefrontApiCustomScalars", {
  enumerable: true,
  get: function () { return hydrogenReact.storefrontApiCustomScalars; }
});
Object.defineProperty(exports, "useLoadScript", {
  enumerable: true,
  get: function () { return hydrogenReact.useLoadScript; }
});
Object.defineProperty(exports, "useMoney", {
  enumerable: true,
  get: function () { return hydrogenReact.useMoney; }
});
Object.defineProperty(exports, "useSelectedOptionInUrlParam", {
  enumerable: true,
  get: function () { return hydrogenReact.useSelectedOptionInUrlParam; }
});
Object.defineProperty(exports, "useShopifyCookies", {
  enumerable: true,
  get: function () { return hydrogenReact.useShopifyCookies; }
});
exports.Analytics = ns;
exports.AnalyticsEvent = M;
exports.CacheCustom = st;
exports.CacheLong = at;
exports.CacheNone = ot;
exports.CacheShort = oe;
exports.CartForm = W;
exports.InMemoryCache = gt;
exports.OptimisticInput = ma;
exports.Pagination = cn;
exports.RichText = cs;
exports.Script = da;
exports.Seo = on;
exports.ShopPayButton = Pa;
exports.VariantSelector = Kn;
exports.cartAttributesUpdateDefault = Vt;
exports.cartBuyerIdentityUpdateDefault = kt;
exports.cartCreateDefault = bt;
exports.cartDiscountCodesUpdateDefault = Ut;
exports.cartGetDefault = Ot;
exports.cartGetIdDefault = $t;
exports.cartGiftCardCodesUpdateDefault = Lt;
exports.cartLinesAddDefault = xt;
exports.cartLinesRemoveDefault = _t;
exports.cartLinesUpdateDefault = Dt;
exports.cartMetafieldDeleteDefault = qt;
exports.cartMetafieldsSetDefault = Ft;
exports.cartNoteUpdateDefault = Nt;
exports.cartSelectedDeliveryOptionsUpdateDefault = Mt;
exports.cartSetIdDefault = Ht;
exports.changelogHandler = An;
exports.createCartHandler = Bt;
exports.createContentSecurityPolicy = sa;
exports.createCustomerAccountClient = wt;
exports.createHydrogenContext = us;
exports.createStorefrontClient = pr;
exports.createWithCache = $o;
exports.formatAPIResult = T;
exports.generateCacheControlHeader = Pe;
exports.getPaginationVariables = pn;
exports.getSelectedProductOptions = zn;
exports.getSeoMeta = nn;
exports.getShopAnalytics = rs;
exports.getSitemap = fs;
exports.getSitemapIndex = ys;
exports.graphiqlLoader = Qo;
exports.storefrontRedirect = Bo;
exports.useAnalytics = j;
exports.useCustomerPrivacy = Wt;
exports.useNonce = Gt;
exports.useOptimisticCart = Hn;
exports.useOptimisticData = fa;
exports.useOptimisticVariant = ea;
